---
title: "Bio3d"
output: html_document
date: "`r Sys.Date()`"
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = TRUE)

library("bio3d")
library("tidyverse")
library("magrittr")
library("data.table")
library("fs")
library("janitor")
#library("rgl")
library("viridis")
library(seqinr)
library("hacksaw")

```

``` {R Try with UCHL1?}

af.uchl1 <- dir_ls("~/Documents/CysteineProfiling/CysteineR/PDB/UCHL1") %>%
  read.pdb()

```

``` {R Useful functions}

get.aa.dist <- function(uniprotid){
  
  print(uniprotid)
  
  n.aa <- af.pred %>%
    filter(protein_id == uniprotid) %>%
    nrow()
  
  if(n.aa != 0){
  
    col.order <- c("protein_id", "position", "AA", 1:n.aa)
    
    af.temp <- af.pred %>%
      filter(protein_id == uniprotid) %>%
      mutate(residue = paste0(AA, position)) %>%
      select(protein_id, residue, x_coord_c, y_coord_c, z_coord_c) %>%
      dist() %>%
      as.matrix() %>%
      as.data.frame() %>% 
      rownames_to_column("position") %>%
      mutate(protein_id = uniprotid) %>%
      left_join(af.residue.match, by = c("protein_id", "position")) %>%
      select(col.order) %>%
      pivot_longer(cols = 4:ncol(.),
                   names_to = "amino.acid2", 
                   values_to = "euc.dist") %>%
      distinct() %>%
      rename(AA.1 = "AA",
             position.1 = position,
             position = "amino.acid2") %>%
      left_join(af.residue.match, by = c("protein_id", "position")) %>%
      rename(AA.2 = "AA", 
             position.2 = "position")
  
  return(af.temp)
    
  }else{
    
    print(paste("Failed on", uniprotid))
  }
  
}

###################

get_group_number = function(){
    i = 0
    function(){
        i <<- i+1
        i
    }
}

group_number = get_group_number()


```

``` {R Junk functions}

find.residue <- function(protein, ## Uniprot ID
                         residue.nr){ ## Residue number from Uniprot annotation
  
  protein <- as.character(protein)
  residue.nr <- as.numeric(residue.nr)
  
  human.fasta[human.fasta$protein_id == protein,] %>%
    pull(Sequence) %>%
    substr(residue.nr, residue.nr)

}

#### JUNK CHUNKS

af.dubs <- dir_ls("/Users/mew21/Documents/GitHub/structuremap/nbs", regexp = "DUBs.csv") %>%
  fread() %>%
  rename(HalfShell = "nAA_12_70_pae",
         WholeShell = "nAA_24_180_pae") %>%
  left_join(gene.protein.id.match)

af.dubs %>%
  group_by(AA) %>%
  mutate(Total = n()) %>%
  ungroup() %>%
  group_by(AA, Total, HalfShell) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Part = n / Total) %>%
  ggplot(aes(x = HalfShell,
             y = Part)) + 
  geom_col() +
  facet_wrap(~AA)

```

```{R 20220714 - Make list of ~10k downloaded AF structures}

cif.dir <- dir_ls("~/Documents/GitHub/structuremap/nbs", regexp = "cif")
pae.dir <- dir_ls("~/Documents/GitHub/structuremap/nbs", regexp = "pae")

cif.files <- dir_ls(cif.dir) %>%
  as.character() 

cif.ids <- cif.files %>%
  sub(pattern = "/Users/mew21/Documents/GitHub/structuremap/nbs/tutorial_cif/(.*).cif", 
      replacement = "\\1")

pae.files <- dir_ls(pae.dir) %>%
  as.character() 

pae.ids <- pae.files %>%
  sub(pattern = "/Users/mew21/Documents/GitHub/structuremap/nbs/tutorial_pae/pae_(.*).hdf", 
      replacement = "\\1")

## Overlap
overlap.ids <- intersect(pae.ids, 
          cif.ids)

## Copy only overlapping files to new dir
overlap.cif.files <- paste0("/Users/mew21/Documents/GitHub/structuremap/nbs/tutorial_cif/", overlap.ids, ".cif")

length(overlap.cif.files)

overlap.pae.files <- paste0("/Users/mew21/Documents/GitHub/structuremap/nbs/tutorial_pae/pae_", overlap.ids, ".hdf")

length(overlap.pae.files)

head(overlap.cif.files)

## Make list of files *NOT* in both dirs
cif.delete <- cif.files[!cif.files %in% overlap.cif.files]

pae.delete <- pae.files[!pae.files %in% overlap.pae.files]

## delete extra cif files
file.remove(cif.delete)


## Make human fasta list with overlap
human.fasta.list <- fread("/Users/mew21/Desktop/HumanFASTA_Headers_AlphaFold.txt") 

human.fasta.overlap <- human.fasta.list %>%
  t() %>%
  as.data.frame() %>%
  filter(V1 %in% overlap.ids)

## Write to .txt
fileConn<-file("HumanFasta_AF_Headers_20220714-Overlap.txt")
writeLines(human.fasta.overlap$V1, fileConn)
close(fileConn)


## Read in human fasta
human.fasta <- dir_ls("/Users/mew21/Library/CloudStorage/OneDrive-ImperialCollegeLondon/DUBs/Resources", regexp = "FASTA") %>%
  fread() %>%
  mutate(protein_id = sub(Gene, 
                          pattern = "..\\|(.*)\\|.*",
                          replacement = "\\1"))
  

########### Re-format human fasta headers to try downloading all in one go
hsap.headers <- human.fasta$protein_id %>%
  paste(collapse = "\t")
  
## Write to .txt
fileConn<-file("/Users/mew21/Downloads/hsapiens_headers.txt")
writeLines(hsap.headers, fileConn)
close(fileConn)

```



``` {R Read in predictions for first 10k}

### Join gene names
gene.protein.id.match <- human.fasta %>%
  mutate(protein_id = sub(Gene, 
                          pattern = "..\\|(.*)\\|.*",
                          replacement = "\\1")) %>%
  select(protein_id, Genes)


af.pred <- dir_ls("/Users/mew21/Documents/GitHub/structuremap/nbs/", regexp = "AlphaFoldPredicted.*hsapiens") %>% 
  fread() %>%
  rename(HalfShell = "nAA_12_70_pae",
         WholeShell = "nAA_24_180_pae")# %>%
#  left_join(gene.protein.id.match)

af.pred %>%
  filter(protein_id == "P00533") %>% View()

```

``` {R Plot distributions of accessibility by AA}

## Background distribution of average for all AAs
af.all.aa.ref <- af.pred %>%
  filter(quality > 80) %>%
  mutate(Total = nrow(.)) %>%
  group_by(Total, HalfShell) %>%
  summarise(n = n(),
            .groups = "drop") %>%
  mutate(Ref.Part = n / Total) %>%
  select(HalfShell, Ref.Part)
         

## Histograms of adjacent AAs (1 / accessibility)
af.by.aa <- af.pred %>%
  filter(quality > 80) %>%
  group_by(AA) %>%
  mutate(Total = n()) %>%
  ungroup() %>%
  group_by(AA, Total, HalfShell) %>%
  summarise(n = n(),
            .groups = "drop") %>%
  mutate(Part = n / Total)

## To rank by number of 0 residues
af.aa.exposed <- af.by.aa %>%
  filter(HalfShell == 0) %>%
  rename(Part.Exposed = "Part") %>%
  select(AA, Part.Exposed) %>%
  distinct()

af.by.aa %>%
  left_join(af.aa.exposed) %>%
  left_join(af.all.aa.ref, by = "HalfShell") %>%
  mutate(Part.Norm = Part - Ref.Part) %>%
  ggplot(aes(x = HalfShell,
             y = Part.Norm)) + 
  geom_col() +
  facet_wrap(~factor(AA, levels = af.aa.exposed %>%
                       arrange(Part.Exposed) %>%
                       pull(AA))) + 
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_rect(fill = "white", colour = "white"),
        strip.text = element_text(colour = "black")) + 
  ylab("% of residues") + 
  xlab("Amino Acid Depth")

#ggsave("AF/Plots/AllAminoAcids_Ordered.png", height = 8, width = 10)

## Bias in quality of prediction by AA?

## Histograms of adjacent AAs (1 / accessibility)
af.pred %>%
  ggplot(aes(x = WholeShell)) + 
  geom_histogram(binwidth = 1) +
  facet_wrap(~AA) + 
  xlim(0,50)

af.pred %>%
  filter(quality > 80) %>%
  ggplot(aes(x = WholeShell,
             y = HalfShell)) + 
  geom_point(alpha = 0.05) + 
  facet_wrap(~AA)
    
    

```

``` {R Compare cysteine and whole proteome distrbutions}
##
af.cys <- af.pred %>%
  filter(AA == "C",
         quality > 80) %>%
  select(protein_id, AA, position, HalfShell) %>%
  mutate(Total = length(AA),
         Class = "Cysteine")

af.ser <- af.pred %>%
  filter(AA == "S",
         quality > 80) %>%
  select(protein_id, AA, position, HalfShell) %>%
  mutate(Total = length(AA),
         Class = "Serine")

af.phe <- af.pred %>%
  filter(AA == "F",
         quality > 80) %>%
  select(protein_id, AA, position, HalfShell) %>%
  mutate(Total = length(AA),
         Class = "Phenylalanine")

af.wp <- af.pred %>%
  select(protein_id, AA, position, HalfShell) %>%
  mutate(Total = length(position),
         Class = "All amino acids")


#### All amino acids vs cysteine

rbind(#af.wp,
      af.ser,
      af.phe,
      af.cys) %>%
  group_by(Class, Total, HalfShell) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Part = n / Total) %>%
  left_join(af.all.aa.ref) %>%
  mutate(Part.Norm = Part - Ref.Part) %>%
  ggplot(aes(x = HalfShell,
             y = 100*Part.Norm,
             fill = Class)) + 
  geom_col(position = position_identity()) +
  facet_wrap(~factor(Class,
                     levels = c(#"All amino acids", 
                                "Serine", "Phenylalanine", "Cysteine")), 
             ncol = 2) + 
  xlab("Amino Acid Depth") + 
  xlim(-0.5,16) +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(colour = "black"),
        legend.position = "none") + 
  ylab("Proportion relative to whole proteome") + 
  scale_y_continuous(expand = expansion(mult = c(0.05,0.05)))

#ggsave("AF/Plots/CysvsAllAA_Distributions.png", height = 4, width = 5)


```


``` {R Cysteine accessibility by function/modification}

### Filter for active site cysteines in Uniprot

### Nucleophiles
uniprot.nucleophile <- uniprot.annotations %>%
  filter(str_detect(`Active site`, "Nucleophile")) %>%
  mutate(position = as.integer(sub(`Active site`, 
                     pattern = "ACT_SITE (\\d*)[^\\d].*",
                     replacement = "\\1"))) %>%
  rename(protein_id = "Entry") %>%
  left_join(af.pred, by = c("protein_id", "position")) %>%
  filter(!is.na(HalfShell)) %>%
  select(protein_id, quality, AA, position, HalfShell) %>%
  filter(AA == "C") %>%
  mutate(Total = length(AA), 
         Class = "Active Site Cys Nucleophile")


#### All active site residues (not just cysteines)
uniprot.activesite <- uniprot.annotations %>%
  filter(!is.na(`Active site`)) %>%
  mutate(position = as.integer(sub(`Active site`, 
                     pattern = "ACT_SITE (\\d*)[^\\d].*",
                     replacement = "\\1"))) %>%
  rename(protein_id = "Entry") %>%
  left_join(af.pred, by = c("protein_id", "position")) %>%
  filter(!is.na(HalfShell)) %>%
  select(protein_id, quality, AA, position, HalfShell) %>%
  filter(AA == "C") %>%
  mutate(Total = length(AA), 
         Class = "All Active Site Residues")


### Palmitoylated cysteines
uniprot.palmitoyl <- uniprot.annotations %>%
  mutate(All.Residues = str_extract_all(Lipidation, 'LIPID (\\d*)[^\\d]* /note="S-palmitoyl cysteine'),
         All.Residues = str_extract_all(as.character(All.Residues), 'LIPID (\\d*)'),
         All.Residues = sub(All.Residues, 
                            pattern = 'c\\(',
                            replacement = ""),
         All.Residues = gsub(All.Residues, 
                            pattern = '[\\(\\;\\"\\)]',
                            replacement = ""),
         All.Residues = gsub(All.Residues, 
                            pattern = 'LIPID ',
                            replacement = "")) %>%
  separate(All.Residues, into = paste("Residue", 1:6), sep = ", ") %>%
  select(c("Entry", "Gene Names", paste("Residue", 1:6))) %>%
  filter(`Residue 1` != "character0") %>%
  pivot_longer(cols = 3:8,
               names_to = "Residue.X",
               values_to = "position") %>%
  select(-Residue.X, -`Gene Names`) %>%
  rename(protein_id = "Entry") %>%
  mutate(position = as.integer(position)) %>%
  left_join(af.pred, by = c("protein_id", "position")) %>%
  filter(!is.na(HalfShell),
         !is.na(position)) %>%
  select(protein_id, quality, AA, position, HalfShell) %>%
  mutate(Total = length(AA), 
         Class = "Palmitoyl")

### Farnesylated proteins
uniprot.prenyl <- uniprot.annotations %>%
  filter(str_detect(Lipidation, 
                    pattern = "farnesyl") | 
           str_detect(Lipidation, pattern = "geranyl")) %>%
  mutate(Residue = sub(Lipidation, 
                     pattern = "LIPID (\\d*)[^\\d].*",
                     replacement = "\\1")) %>%
  rename(protein_id = "Entry") %>%
  left_join(human.fasta) %>%
  mutate(AA = substr(Sequence, start = Residue, stop = Residue)) %>%
  rename(position = "Residue") %>%  
  mutate(position = as.integer(position)) %>%
  left_join(af.pred, by = c("protein_id", "position")) %>%
  select(protein_id,quality,  AA.x, position, HalfShell) %>%
  rename(AA = "AA.x") %>%
  filter(!is.na(HalfShell),
         AA == "C") %>%
  mutate(Total = length(AA), 
         Class = "Prenyl")

## Metal binding cysteines
## TODO -- find the specific metals bound -- LATER
uniprot.metalcys <- uniprot.annotations %>%
  filter(!is.na(`Metal binding`)) %>%
  mutate(All.Residues = str_extract_all(`Metal binding`, "METAL (\\d*)[^\\d]"),
         #All.Metals = str_extract_all(`Metal binding`, 'note=\\".*\\"\\; \\/')) %>%
         All.Residues = gsub(All.Residues, 
                            pattern = '[c\\(\\;\\"\\)]',
                            replacement = "")) %>%
  separate(All.Residues, into = paste("Residue", 1:35), sep = ", ") %>%
  select(c("Entry", "Gene Names", paste("Residue", 1:35))) %>%
  filter(`Residue 1` != "harater0") %>%
  pivot_longer(cols = 3:37,
               names_to = "Residue.X",
               values_to = "Residue") %>%
  filter(!is.na(Residue)) %>%
  select(-Residue.X) %>%
  mutate(position = as.integer(sub(Residue, 
                       pattern = "METAL ",
                       replacement = ""))) %>%
  rename(protein_id = "Entry") %>%
  select(protein_id, position) %>%
  left_join(af.pred, by = c("protein_id", "position")) %>%
  filter(!is.na(HalfShell)) %>%
  select(protein_id,quality,  AA, position, HalfShell) %>%
  filter(AA == "C") %>%
  mutate(Total = length(AA), 
         Class = "Metal Binding") 

## Disulfides
uniprot.disulfide <- uniprot.annotations %>%
  filter(str_detect(`Disulfide bond`, "DISULF")) %>%
  select(Entry, `Gene Names`, `Disulfide bond`) %>%
  mutate(Start = sub(`Disulfide bond`, 
                     pattern = "DISULFID (\\d*)[^\\d].*",
                     replacement = "\\1"),
         End = sub(`Disulfide bond`, 
                     pattern = "DISULFID \\d*\\.\\.(\\d*).*",
                     replacement = "\\1")) %>%
  pivot_longer(cols = 4:5, 
               names_to = "Type",
               values_to = "Residue") %>%
  select(-Type) %>%
  rename(protein_id = "Entry") %>%
  left_join(human.fasta) %>%
  mutate(AA = substr(Sequence, start = Residue, stop = Residue)) %>%
  rename(position = "Residue") %>% 
  mutate(position = as.integer(position)) %>%
  left_join(af.pred, by = c("protein_id", "position")) %>%
  rename(AA = "AA.x") %>%
  filter(!is.na(HalfShell),
         !is.na(AA)) %>%
  select(protein_id, quality, AA, position, HalfShell) %>%
  mutate(Total = length(AA), 
         Class = "Disulfide") %>%
  filter(AA == "C")

## ALl cysteines across proteome
human.cys.wp <- af.pred %>%
  filter(AA == "C") %>%
  select(protein_id, quality, AA, position, HalfShell) %>%
  mutate(Total = length(AA),
         Class = "Whole Proteome") %>%
  group_by(AA, #log.intensity, 
           Total, HalfShell) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Part = n / Total) 

human.cys.ref.dist <- af.pred %>%
  filter(AA == "C",
         #quality > 80
         ) %>% ## Only very confident cys predictions
  select(protein_id, AA, position, HalfShell) %>%
  mutate(Total = length(AA),
         Class = "Whole Proteome") %>%
  group_by(AA, #log.intensity, 
           Total, HalfShell) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Ref.Part = n / Total) %>%
  select(HalfShell, Ref.Part)

rbind(#af.cys.wp,
      uniprot.activesite,
      uniprot.nucleophile,
      uniprot.palmitoyl,
      uniprot.prenyl,
      uniprot.disulfide,
      uniprot.metalcys) %>% 
  #filter(quality > 80) %>%
  group_by(AA, Class, Total, HalfShell) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Part = n / Total) %>%
  left_join(human.cys.ref.dist, by = "HalfShell") %>% 
  mutate(Part.Norm = Part - Ref.Part) %>%
  ggplot(aes(x = HalfShell,
             y = 100*Part.Norm,
             fill = Class)) + 
  geom_col(position = position_identity()) +
  facet_wrap(~factor(Class),
                     #levels = c("Whole Proteome", "Disulfide", "Active Site Nucleophile")), 
             ncol = 3) + 
  xlab("Amino Acid Depth") + 
  #xlim(-0.5,16) +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(colour = "black"),
        legend.position = "none") + 
  ylab("% enrichment vs. whole proteome") + 
  scale_y_continuous(expand = expansion(mult = c(0.05,0.05)))

ggsave("AF/Plots/CysPTM_Distributions_ActiveSite.png", height = 2, width = 5)



########################################


rbind(af.cys.wp,
      uniprot.palmitoyl,
      uniprot.prenyl) %>%
  group_by(AA, Class, Total, HalfShell) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Part = n / Total) %>%
  arrange(Class) %>%
  ggplot(aes(x = HalfShell,
             y = 100*Part,
             fill = Class)) + 
  geom_col(position = position_identity()) +
  facet_wrap(~factor(Class,
                     levels = c("Whole Proteome", "Palmitoyl", "Prenyl")), 
             ncol = 3) + 
  xlab("Amino Acid Depth") + 
  xlim(-0.5,16) +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(colour = "black"),
        legend.position = "none") + 
  ylab("% of amino acids") + 
  scale_y_continuous(expand = expansion(mult = c(0.05,0.05)))

ggsave("AF/Plots/CysPTM_Distributions_Lipid.png", height = 2, width = 5)

########################################


 rbind(af.cys.wp,
      uniprot.activesite,
      uniprot.nucleophile,
      uniprot.disulfide) %>%
  group_by(AA, Class, Total, HalfShell) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Part = n / Total) %>%
  ungroup() %>%
  group_by(Class) %>%
  summarise(sum = sum(Part))


```

``` {R More detail on palmitoylated proteins}
## Is zDHHC substrate/residue choice dictated by membrane proximity and cysteine accessibility alone?

## Palmitoylated residue list
palm.residues <- paste(uniprot.palmitoyl$protein_id, 
                       uniprot.palmitoyl$position)

## Get list of proteins with at least 1 annotated palmitoyl site
palm.proteins <- af.pred %>%
  filter(protein_id %in% uniprot.palmitoyl$protein_id,
         AA == "C") %>%
  mutate(join = paste(protein_id, position),
         Acylated = ifelse(join %in% palm.residues, 
                           yes = "Palmitoylated",
                           no = "Non-palmitoylated")) %>%
  group_by(Acylated) %>%
  mutate(Total = n()) %>%
  ungroup()

unique(palm.proteins$protein_id) %>% length()

palm.proteins %>%
  select(protein_id, position, Total, HalfShell, Acylated) %>%
  distinct() %>%
  group_by(HalfShell, Total, Acylated) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Part = n / Total) %>%
  ggplot(aes(x = HalfShell,
             y = 100*Part,
             fill = Acylated)) + 
  geom_col(position = position_identity()) +
  facet_wrap(~factor(Acylated)) + 
  xlab("Amino Acid Depth") + 
  #xlim(-0.5,16) +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(colour = "black", hjust = -0.03, size = 11, face = "bold"),
        legend.position = "none") + 
  ylab("% of amino acids") + 
  scale_y_continuous(expand = expansion(mult = c(0,0.05))) +
  ggtitle("Accessibility of cysteine residues in palmitoylated proteins\nby acylation status (284 proteins, 284 annotated acylation sites)")

ggsave("plots/PalmitoylatedCysteines.png", height = 4, width = 6.5)

### TFR1 
af.pred %>%
  filter(protein_id == "P02786",
         AA == "C") %>% 
  select(AA, position, secondary_structure, structure_group, HalfShell) %>% View

```

``` {R Download full UniProt annotation}

uniprot.annotations <- dir_ls("~/Downloads", regexp = "allann") %>%
  fread()


## Compare to active site + whole proteome


str(af.pred)
str(uniprot.disulfide)

```

``` {R Public data solvent exposure distribution}

### Gygi / Cravatt nature biotech 2021
gygi.df <- dir_ls("~/Downloads", regexp = "41587_2020_778_MOESM9_ESM")[1] %>%
  readxl::read_excel(sheet = 2) %>%
  mutate(protein_id = sub(`Uniprot ID`, 
                           pattern = "..\\|(.*)\\|.*",
                           replacement = "\\1")) %>%
  rename(position = "Site Position") %>%
  left_join(af.pred) %>%
  select(protein_id, AA, position, HalfShell) %>%
  distinct() %>%
  filter(!is.na(HalfShell)) %>%  
  mutate(Total = length(AA),
         Dataset = "SLC-ABPP")


### Thiopyridyl agarose
tp.resin.df <- dir_ls("/Users/mew21/Library/CloudStorage/OneDrive-ImperialCollegeLondon/Proteomics/LMS/20220425_TP-Resin_DDA/b008p107/MSFragger/OpenSearch/20220608_Cys_Open",
       regexp = "peptide.tsv") %>%
  fread() %>% 
  mutate(Peptide.Cys = str_locate(Peptide, ## Location of Cys in peptide
                                  "C")[,1] - 2) %>% # account for +1 of matching
  rename(protein_id = "Protein ID") %>%
  left_join(human.fasta, by = "protein_id") %>% ## join whole protein sequence from FASTA 
  mutate(Peptide.Start = str_locate(string = Sequence, ## find peptide location in protein
                                    pattern = Peptide)[,1], ## [1] takes start of seq
         position = Peptide.Start + Peptide.Cys + 1) %>%
  left_join(af.pred, by = c("protein_id", "position")) %>%
  select(protein_id, AA, position, HalfShell) %>%
  distinct() %>%
  filter(!is.na(HalfShell)) %>%
  mutate(Total = length(AA),
         Dataset = "TP-Resin")

### First DBIA experiment
dbia.df <- dir_ls("/Users/mew21/Library/CloudStorage/OneDrive-ImperialCollegeLondon/CysteineProfiling/DBIA/Data/20220630_tripletof_dbia_testinjections", regexp = "20220702_DBIA_Fixed296_noIAA_EditLib_Structural_Strip_PredictedfromFASTA_Output.tsv") %>%
  fread() %>%
  filter(str_detect(Run,
                    "20220701_DBIA")) %>%
  mutate(protein_id = sub(Protein.Group,
                          pattern = "(.*)\\;.*",
                          replacement = "\\1"),
         Peptide.Cys = str_locate(Stripped.Sequence, ## Location of Cys in peptide
                                  "C")[,1] - 2) %>% # account for +1 of matching
  left_join(human.fasta, by = "protein_id") %>% ## join whole protein sequence from FASTA 
  mutate(Peptide.Start = str_locate(string = Sequence, ## find peptide location in protein
                                    pattern = Stripped.Sequence)[,1], ## [1] takes start of seq
         position = Peptide.Start + Peptide.Cys + 1) %>%
  left_join(af.pred, by = c("protein_id", "position")) %>%
  select(protein_id, AA, position, HalfShell) %>%
  distinct() %>%
  filter(!is.na(HalfShell)) %>%
  mutate(Total = length(AA),
         Dataset = "DBIA")

### Weerapana caged probes -- in living cells
caged.live.df <- dir_ls("~/Downloads/", regexp = "/ja5b04350_si_001") %>%
  readxl::read_excel(sheet = 3) %>%
  row_to_names(row_number = 1) %>%
  rename(protein_id = "ID", 
         position = "Cysteine position") %>%
  mutate(position = as.integer(position)) %>%
  left_join(af.pred, by = c("protein_id", "position")) %>%
  select(protein_id, AA, position, HalfShell) %>%
  distinct() %>%
  filter(!is.na(HalfShell)) %>%
  mutate(Total = length(AA),
         Dataset = "Caged probe,\nlive cells")

### Weerapana caged probes -- lysates
caged.lysate.df <- dir_ls("~/Downloads/", regexp = "/ja5b04350_si_001") %>%
  readxl::read_excel(sheet = 1) %>%
  row_to_names(row_number = 1) %>%
  rename(protein_id = "ID", 
         position = "Cysteine position") %>%
  mutate(position = as.integer(position)) %>%
  left_join(af.pred, by = c("protein_id", "position")) %>%
  select(protein_id, AA, position, HalfShell) %>%
  distinct() %>%
  filter(!is.na(HalfShell)) %>%
  mutate(Total = length(AA),
         Dataset = "Caged probe,\nlysed cells")

rbind(af.cys %>% mutate(Dataset = "Whole\nProteome") %>% select(-Class),
      tp.resin.df,
      dbia.df#,
      #gygi.df,
      #caged.live.df,
      #caged.lysate.df
      ) %>%
  group_by(AA, Dataset, Total, HalfShell) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Part = n / Total) %>%
  arrange(Dataset) %>%
  ggplot(aes(x = HalfShell,
             y = Part,
             fill = Dataset)) + 
  geom_col(position = position_identity()) +
  facet_wrap(~Dataset, ncol = 1,
             strip.position = "right") +
  xlab("Amino Acid Depth") + 
  xlim(-0.5,16) +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(colour = "black"),
        strip.text.y.right = element_text(angle = 0),
        legend.position = "none") + 
  ylab("% of amino acids") + 
  scale_y_continuous(expand = expansion(mult = c(0,0.05)))

ggsave("AF/Plots/CysProfilingMine_Distributions.png", height = 6, width = 4)

  
############################################################

rbind(af.cys %>% mutate(Dataset = "Whole\nProteome") %>% select(-Class),
      #tp.resin.df,
      #dbia.df#,
      #gygi.df,
      caged.live.df,
      caged.lysate.df
      ) %>%
  group_by(AA, Dataset, Total, HalfShell) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Part = n / Total) %>%
  arrange(Dataset) %>%
  ggplot(aes(x = HalfShell,
             y = Part,
             fill = Dataset)) + 
  geom_col(position = position_identity()) +
  facet_wrap(~Dataset, ncol = 1,
             strip.position = "right") +
  xlab("Amino Acid Depth") + 
  xlim(-0.5,16) +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(colour = "black"),
        strip.text.y.right = element_text(angle = 0),
        legend.position = "none") + 
  ylab("% of amino acids") + 
  scale_y_continuous(expand = expansion(mult = c(0,0.05)))

ggsave("AF/Plots/CysProfilingPublished_Distributions.png", height = 6, width = 4)



```

``` {R DBIA Enrichment}

dbia.enrichment <- dir_ls("/Users/mew21/Library/CloudStorage/OneDrive-ImperialCollegeLondon/CysteineProfiling/DBIA/Data/20220630_tripletof_dbia_testinjections", regexp = "20220702_DBIA_Fixed296_noIAA_EditLib_Structural_Strip_PredictedfromFASTA_Output.tsv") %>%
  fread() %>%
  filter(str_detect(Run,
                    "20220701_DBIA")) %>%
  mutate(Contains.Cys = ifelse(str_detect(Stripped.Sequence, "C"),
                               yes = "Yes",
                               no = "No"))


dbia.enrichment %>%
  select(Stripped.Sequence, Contains.Cys) %>%
  distinct() %>%
  group_by(Contains.Cys) %>%
  summarise(n=n())



```

``` {R Compute 3D distance from Cys to nearby AA}

af.residue.match <- af.pred %>%
  select(protein_id, position, AA) %>%
  distinct()

#af.residue.match$position <- as.character(af.residue.match$position)

rm(af.temp)

## Look at UCHL1
af.uchl1 <- get.aa.dist("P09936") %>%
  filter(AA.1 == "C")

af.uchl1 %>%
  mutate(aa.dist = abs(as.numeric(position.1) - as.numeric(position.2))) %>%
  ggplot(aes(x = aa.dist, 
             y = euc.dist,
             colour = position.1)) + 
  geom_point(size = 0.25) + 
  facet_wrap(~position.1)


#### Rab27A -- cluster useful cysteines?
af.rab27a <- get.aa.dist("P51159") %>%
  filter(AA.1 == "C")

af.rab27a %>%
  mutate(aa.dist = as.numeric(position.1) - as.numeric(position.2)) %>%
  ggplot(aes(x = aa.dist, 
             y = euc.dist,
             colour = position.1)) + 
  geom_point(size = 0.25) + 
  facet_wrap(~position.1)

#### ACAT1- from gygi paper - look at different cysteines in close proximity
af.acat1 <- get.aa.dist("P24752") %>%
  filter(AA.1 == "C")

af.acat1 %>%
  mutate(aa.dist = as.numeric(position.1) - as.numeric(position.2),
         centred.dist = aa.dist - as.numeric(position.1)) %>%
  ggplot(aes(x = centred.dist, 
             y = euc.dist,
             colour = position.1)) + 
  geom_point(size = 0.25) + 
  facet_wrap(~position.1)

af.acat1 %>%
  mutate(aa.dist = as.numeric(position.1) - as.numeric(position.2),
         centred.dist = aa.dist - as.numeric(position.1)) %>%
  select(position.1, position.2, euc.dist) %>%
  pivot_wider(values_from = "euc.dist",
              names_from = "position.1") %>%
  column_to_rownames("position.2") %>%
  cor()

## Can we use the pairwise distances -- i.e. shape of curve to cluster cysteines structurally?
## Plot scatter of centred distances... Or does this just give us the position back?

## Use this approach to find distal amino acids with very similar positions? Number of nearby but non-adjacent residues (of catalytic activity...?) indicates pocket /  
## Use this approach to compare residues from similar classes of proteins? 


```

``` {R Try to compute correlations between different cysteines from DUBs}

dub.list <- dub.annotation %>%
  rename(Genes = "Gene names") %>%
  left_join(gene.protein.id.match) %>%
  pull(protein_id)

dub.to.predict <- intersect(dub.list,
          af.pred$protein_id)

af.dub.rhyme <- data.frame()

for(i in seq_along(dub.to.predict)){
  
  
  
  df.temp <- get.aa.dist(dub.to.predict[i]) %>%
  filter(AA.1 == "C") %>%
  mutate(aa.dist = as.numeric(position.1) - as.numeric(position.2),
         centred.dist = aa.dist - as.numeric(position.1)) 

  af.dub.rhyme <- rbind(af.dub.rhyme, df.temp)
  
}


af.dub.rhyme %>%
  left_join(gene.protein.id.match) %>%
  filter(str_detect(Genes, "OTUB")) %>%
  ggplot(aes(x = centred.dist, 
             y = euc.dist,
             colour = position.1)) + 
  geom_line(size = 0.25) + 
  facet_wrap(~paste(Genes)) + 
  theme(legend.position = "none")

### Compute correlations across all DUB cysteines
af.dub.cor <- af.dub.rhyme %>%
  left_join(gene.protein.id.match) %>%
  mutate(gene.position1 = paste0(Genes,"-C",position.1)) %>%
  select(gene.position1, position.2, euc.dist) %>%
  distinct() %>%
  pivot_wider(values_from = "euc.dist",
              names_from = "gene.position1") %>% {
                
                as.matrix(af.)
                
              }
  column_to_rownames("position.2") %>%
  cor()


```

``` {R Compute pI prediction for dbia peptides (ballpark)}
library(Peptides)

dbia.enrichment %>%
  select(Stripped.Sequence) %>%
  mutate(pI = Peptides::pI(Stripped.Sequence)) %>% View

```

``` {R S aureus headers for structuremap}

### Read in S. aureus fasta
library(seqinr)

saur.fasta <- dir_ls("/Users/mew21/Library/CloudStorage/OneDrive-ImperialCollegeLondon/CysteineAccessibility/saureus/fasta", regexp = "\\.fasta") %>%
  read.fasta()

saur.fasta.df <- do.call(rbind.data.frame, saur.fasta)

saur.headers <- names(saur.fasta) %>%
  sub(pattern = "..\\|(.*)\\|.*",
      replacement = "\\1") %>%
  paste(collapse = "\t")

## Write to .txt
fileConn<-file("/Users/mew21/Downloads/saureus_headers.txt")
writeLines(saur.headers, fileConn)
close(fileConn)

write.csv(saur.headers,
          "~/Downloads/saureus_headers.csv",
          quote = F,
          col.names = F,
          row.names = F)


## Read in human fasta
human.fasta <- dir_ls("/Users/mew21/Library/CloudStorage/OneDrive-ImperialCollegeLondon/DUBs/Resources", regexp = "FASTA") %>%
  fread() %>%
  mutate(protein_id = sub(Gene, 
                          pattern = "..\\|(.*)\\|.*",
                          replacement = "\\1"))
  


write_csv(af.dubs %>%
            select(protein_id, Genes, AA, position, 
                   x_coord_ca, y_coord_ca, z_coord_ca,
                   HalfShell, WholeShell),
          "~/Downloads/AlphaFold_DUBs_Accessiblity.csv")


```

``` {R Mus musculus headers -- for oxiMouse}

mus.fasta.raw <- dir_ls("/Users/mew21/Library/CloudStorage/OneDrive-ImperialCollegeLondon/CysteineAccessibility/musmus/fasta", regexp = ".fasta") %>%
  read.fasta(seqtype = "AA",
             as.string = T,
             whole.header = T)



mus.headers <- names(mus.fasta) %>%
  sub(pattern = "..\\|(.*)\\|.*",
      replacement = "\\1") %>%
  paste(collapse = "\t")


## Write to .txt
fileConn<-file("/Users/mew21/Downloads/musmus_headers.txt")
writeLines(mus.headers, fileConn)
close(fileConn)



```


##############################################################################################

### TODO: 
# Global analysis of different organisms and accessibility (more a reflection of AF predictions?)
# 

## Analysis of Zanon et al. (ChemRxiv 2021) Cys-reactive probes

``` {R Read in PZ supplementary table data}

## Choose cys-reactive probes to read in and check distributions
pz.cys.probes <- c("IA", "BMK", "MSBT", "MST", "MSOD", "EBX1", "EBX2")

## Read in supplementary tables
pz.supp.sheets <- dir_ls("zanon_raw/", regexp = "table-4") %>% ## + Quant
  readxl::excel_sheets()

pz.cys.probes.toread <- pz.supp.sheets[pz.supp.sheets %in% paste0("SA ", pz.cys.probes, "-alkyne")]

pz.cys.raw <- data.frame()

for(i in seq_along(pz.cys.probes.toread)){

  pz.temp <- dir_ls("zanon_raw/", regexp = "table-4") %>%
    readxl::read_excel(sheet = pz.cys.probes.toread[i]) %>%
    mutate(Probe = sub(pz.cys.probes.toread[i],
                       pattern = "SA (.*)-alkyne",
                       replacement = "\\1"))
  
  pz.cys.raw <- rbind(pz.cys.raw, 
                  pz.temp)
  
}

rm(pz.temp)


```

``` {R Reformat and join AF / sequence info to cys-proteomic data}

pz.cys.df <- pz.cys.raw %>%
  rename(protein_id = "UniProt code") %>%
  left_join(saur.fasta, by = "protein_id") %>%
  filter(!is.na(Gene)) %>%
  mutate(Peptide.Cys = str_locate(string = `Modified Peptide`,
                                  pattern = "\\*")[,1] - 2,
         Peptide.Start = str_locate(string = Sequence, ## find peptide location in protein
                                    pattern = `Peptide Sequence`)[,1], ## [1] takes start of seq
         position = Peptide.Start + Peptide.Cys) %>%
  mutate(log.intensity = log10(`Total Intensity`),
         intensity_decile = cut(log.intensity, breaks=10),
         intensity_decile = sub(intensity_decile, 
                                pattern = "\\((.*)\\,.*",
                                replacement = "\\1"),
         rank_decile = dense_rank(desc(intensity_decile))) %>%
  ungroup() %>%
  select(-Peptide.Cys, -Gene, -Peptide.Start) %>%
  distinct() %>%
  select(protein_id, `Essential?`, log.intensity, Probe, position) %>%
  left_join(af.saur, by = c("protein_id", "position")) %>%
  group_by(Probe) %>%
  mutate(Total = n()) %>% ## Total cysteines per probe (for %)
  ungroup() 


### Plotting by different probes
pz.cys.df %>%
 #filter(Probe == "IA") %>%
  group_by(AA, Probe, #log.intensity, 
           Total, HalfShell) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Part = n / Total) %>%
  arrange(Probe) %>%
  ggplot(aes(x = HalfShell,
             y = Part, 
             fill = Probe)) + 
  geom_col() +
  #geom_density_2d_filled(n=50) +
  facet_wrap(~Probe, 
             ncol = 2) +
  xlab("Amino Acid Depth") + 
  xlim(-0.5,16) +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(colour = "black", hjust = -0.01),
        strip.text.y.right = element_text(angle = 0),
        legend.position = "none") + 
  ylab("% of amino acids") + 
  scale_y_continuous(expand = expansion(mult = c(0,0.05)))

saur.cys.wp <- af.saur %>%
  filter(AA == "C") %>%
  select(protein_id, AA, position, HalfShell) %>%
  mutate(Total = length(AA),
         Class = "Whole Proteome") %>%
  group_by(AA, #log.intensity, 
           Total, HalfShell) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Part = n / Total) 

saur.lys.wp <- af.saur %>%
  filter(AA == "K") %>%
  select(protein_id, AA, position, HalfShell) %>%
  mutate(Total = length(AA),
         Class = "Whole Proteome") %>%
  group_by(AA, #log.intensity, 
           Total, HalfShell) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Part = n / Total) 

saur.cys.wp %>%
  ggplot(aes(x = HalfShell,
             y = Part)) + 
  geom_col() +
  #facet_wrap(~Probe, 
           #  ncol = 2) +
  xlab("Amino Acid Depth") + 
  xlim(-0.5,16) +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(colour = "black", hjust = -0.01),
        strip.text.y.right = element_text(angle = 0),
        legend.position = "none") + 
  ylab("% of amino acids") + 
  scale_y_continuous(expand = expansion(mult = c(0,0.05)))



```

``` {R Compare human, mouse and s.aureus accessibilities}

colnames(mus.cys.wp)
colnames(human.cys.wp) 
colnames(saur.cys.wp)

organism.cys <- rbind(mus.cys.wp %>%
                        mutate(Organism = "Mouse"),
                      human.cys.wp %>%
                        mutate(Organism = "Human"),
                      saur.cys.wp %>%
                        mutate(Organism = "S. aureus")) 

organism.cys %>%
  mutate(facet = paste0(Organism, " (", Total, " cysteines", ")")) %>%
  ggplot(aes(x = HalfShell,
             y = Part)) + 
  geom_col() +
  facet_wrap(~facet,
             ncol = 1) +
  xlab("Amino Acid Depth") + 
  xlim(-0.5,16) +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(colour = "black", hjust = -0.01),
        strip.text.y.right = element_text(angle = 0),
        legend.position = "none") + 
  ylab("% of amino acids") + 
  scale_y_continuous(expand = expansion(mult = c(0,0.05)))

ggsave("plots/3OrganismsCysAccess.png", height = 7, width = 3)

prediction.quality <- rbind(af.pred %>%
                              filter(AA == "C") %>%
                              mutate(Organism = "Human"),
                            af.saur %>%
                              filter(AA == "C") %>%
                              mutate(Organism = "S. aureus"),
                            af.mus %>%
                              filter(AA == "C") %>%
                              mutate(Organism = "Mouse"))

unique(prediction.quality$structure_group)

prediction.quality %>%
  ggplot(aes(x = quality,
             fill = Organism),
         alpha = 0.5) + 
  geom_density() + 
  facet_grid(structure_group~Organism) +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(colour = "black", hjust = -0.01),
        strip.text.y.right = element_text(angle = 0),
        legend.position = "none") + 
  ylab("AlphaFold prediction quality") + 
  scale_y_continuous(expand = expansion(mult = c(0,0.05)))

ggsave("plots/PredictionQuality_Organisms.png", height = 7, width = 6)

```

``` {R Plot accessibilities by AF prediction confidence}

colnames(af.pred)

min(af.pred$quality)

af.pred %>%
  filter(AA == "C") %>%
  select(protein_id, AA, position, HalfShell, structure_group) %>%
  mutate(Total = length(AA),
         Class = "Whole Proteome") %>%
  group_by(structure_group, AA, #log.intensity, 
           Total, HalfShell) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Part = n / Total) %>%
  ggplot(aes(x = HalfShell,
             y = 100*Part)) + 
  geom_col() + 
  facet_wrap(~structure_group) +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(colour = "black", hjust = -0.01),
        strip.text.y.right = element_text(angle = 0),
        legend.position = "none") + 
  ylab("% of residues") + 
  scale_y_continuous(expand = expansion(mult = c(0,0.05))) + 
  xlab("Amino Acid Depth")



``` 
``` {R Bias for structure groups in Gygi data}

### Gygi / Cravatt nature biotech 2021
gygi.structure <- dir_ls("~/Downloads", regexp = "41587_2020_778_MOESM9_ESM")[1] %>%
  readxl::read_excel(sheet = 2) %>%
  mutate(protein_id = sub(`Uniprot ID`, 
                           pattern = "..\\|(.*)\\|.*",
                           replacement = "\\1")) %>%
  rename(position = "Site Position") %>%
  left_join(af.pred) %>%
  select(protein_id, structure_group, AA, position, HalfShell) %>%
  distinct() %>%
  filter(!is.na(HalfShell)) %>% 
  mutate(Probe = "IA-DTB",
         Total = length(AA))

gygi.df %>%
  select(protein_id, structure_group, AA, position, HalfShell) %>%
  distinct() %>%
  filter(!is.na(HalfShell)) %>%  
  mutate(Total = length(AA),
         Dataset = "SLC-ABPP") %>%
  group_by(structure_group, AA, #log.intensity, 
           Total, HalfShell) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Part = n / Total) %>%
  ggplot(aes(x = HalfShell,
             y = 100*Part)) + 
  geom_col() + 
  facet_wrap(~structure_group) +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(colour = "black", hjust = -0.01),
        strip.text.y.right = element_text(angle = 0),
        legend.position = "none") + 
  ylab("% of residues") + 
  scale_y_continuous(expand = expansion(mult = c(0,0.05)))

##### Caged probes?
### Weerapana caged probes -- in living cells
caged.live.structure <- dir_ls("~/Downloads/", regexp = "/ja5b04350_si_001") %>%
  readxl::read_excel(sheet = 3) %>%
  row_to_names(row_number = 1) %>%
  rename(protein_id = "ID", 
         position = "Cysteine position") %>%
  mutate(position = as.integer(position)) %>%
  left_join(af.pred, by = c("protein_id", "position")) %>%
  select(protein_id, structure_group, AA, position, HalfShell) %>%
  distinct() %>%
  filter(!is.na(HalfShell)) %>% 
  mutate(Probe = "Photocaged",
         Total = length(AA))


colnames(caged.live.structure)
colnames(gygi.structure)

rbind(caged.live.structure,
      gygi.structure) %>%
  select(protein_id, structure_group, AA, Total, position, Probe, HalfShell) %>%
  distinct() %>%
  filter(!is.na(HalfShell)) %>% 
  group_by(structure_group, Probe, AA, 
           Total, HalfShell) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Part = n / Total) %>%
  ggplot(aes(x = HalfShell,
             y = 100*Part)) + 
  geom_col() + 
  facet_grid(Probe~structure_group) +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(colour = "black", hjust = -0.01),
        strip.text.y.right = element_text(angle = 0),
        legend.position = "none") + 
  ylab("% of residues") + 
  scale_y_continuous(expand = expansion(mult = c(0,0.05))) + 
  xlab("Amino Acid Depth")

ggsave("plots/GygiCaged_ByStructure.png", height = 4.5, width = 8)


```

``` {R Same with selected lysine probes now -- compare}

## Choose cys-reactive probes to read in and check distributions
pz.lys.probes <- c("STP", "TFP", "NHS", "ATT", "NASA", "ArSq", "EBA")

## Read in supplementary tables
pz.supp.sheets <- dir_ls("zanon_raw/", regexp = "table-4") %>% ## + Quant
  readxl::excel_sheets()

pz.lys.probes.toread <- pz.supp.sheets[pz.supp.sheets %in% paste0("SA ", pz.lys.probes, "-alkyne")]

pz.lys.raw <- data.frame()

for(i in seq_along(pz.lys.probes.toread)){

  pz.temp <- dir_ls("zanon_raw/", regexp = "table-4") %>%
    readxl::read_excel(sheet = pz.lys.probes.toread[i]) %>%
    mutate(Probe = sub(pz.lys.probes.toread[i],
                       pattern = "SA (.*)-alkyne",
                       replacement = "\\1"))
  
  pz.lys.raw <- rbind(pz.lys.raw, 
                  pz.temp)
  
}

rm(pz.temp)


```

``` {R Find lysine residue numbers, join AF data and plot as before}

pz.lys.df <- pz.lys.raw %>%
  rename(protein_id = "UniProt code") %>%
  left_join(saur.fasta, by = "protein_id") %>%
  filter(!is.na(Gene)) %>%
  mutate(Peptide.Lys = str_locate(string = `Modified Peptide`,
                                  pattern = "\\*")[,1] - 2,
         Peptide.Start = str_locate(string = Sequence, ## find peptide location in protein
                                    pattern = `Peptide Sequence`)[,1], ## [1] takes start of seq
         position = Peptide.Start + Peptide.Lys) %>%
  mutate(log.intensity = log10(`Total Intensity`),
         intensity_decile = cut(log.intensity, breaks=10),
         intensity_decile = sub(intensity_decile, 
                                pattern = "\\((.*)\\,.*",
                                replacement = "\\1"),
         rank_decile = dense_rank(desc(intensity_decile))) %>%
  ungroup() %>%
  select(-Peptide.Lys, -Gene, -Peptide.Start) %>%
  distinct() %>%
  select(protein_id, `Essential?`, log.intensity, Probe, position) %>%
  left_join(af.saur, by = c("protein_id", "position")) %>%
  group_by(Probe) %>%
  mutate(Total = n()) %>% ## Total cysteines per probe (for %)
  ungroup() 

### Plotting by different probes
pz.lys.df %>%
  group_by(AA, Probe, Total, HalfShell) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Part = n / Total) %>%
  arrange(Probe) %>%
  rbind(saur.lys.wp %>%
          mutate(Probe = "Whole proteome")) %>%
  ggplot(aes(x = HalfShell,
             y = 100*Part,
             fill = Probe)) + 
  geom_col(position = position_identity()) +
  facet_wrap(~Probe, 
             ncol = 2) +
  xlab("Amino Acid Depth") + 
  xlim(-0.5,16) +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(colour = "black", hjust = -0.01),
        strip.text.y.right = element_text(angle = 0),
        legend.position = "none") + 
  ylab("% of amino acids") + 
  scale_y_continuous(expand = expansion(mult = c(0,0.05)))

ggsave("plots/Saur_LysProbes.png", height = 7, width = 7)


```

``` {R Plot some Cys and Lys probes b2b}

rbind(pz.cys.df %>%
        filter(Probe %in% c("IA", "EBX2"),
               AA == "C") %>%
        select(c(colnames(saur.cys.wp), "Probe")),
      saur.cys.wp %>% 
        mutate(Probe = "All cys"),
      pz.lys.df %>%
        filter(Probe %in% c("NHS", "STP"),
               AA == "K") %>%
        select(c(colnames(saur.cys.wp), "Probe")),
      saur.lys.wp %>%
        mutate(Probe = "All lys")) %>%
  group_by(AA, Probe, #log.intensity, 
           Total, HalfShell) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Part = n / Total) %>%
  arrange(Probe) %>%
  ggplot(aes(x = HalfShell,
            # y = log.intensity,
             y = Part)) + 
  geom_col() +
  #geom_density_2d_filled(n=50) +
  facet_wrap(~paste0("Targeted residue: ", AA,", Probe: ", Probe), 
             ncol = 2) +
  xlab("Amino Acid Depth") + 
  xlim(-0.5,16) +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(colour = "black"),
        strip.text.y.right = element_text(angle = 0),
        legend.position = "none") + 
  ylab("log10(Peptide intensity)") + 
  scale_y_continuous(expand = expansion(mult = c(0,0.05)))

#rbind(
pz.cys.df %>%
        #filter(Probe %in% c("IA", "EBX2"),
        #       AA == "C"),
      #pz.lys.df %>%
      #  filter(Probe %in% c("NHS", "STP"),
      #         AA == "K")) %>%
  group_by(AA, Probe, log.intensity, Total, HalfShell) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Part = n / Total) %>%
  arrange(Probe) %>%
  ggplot(aes(x = HalfShell,
             y = log.intensity)) + 
  #geom_point(size = 5,
  #           shape = 15) +
  geom_density_2d_filled(n=50) +
  facet_wrap(~Probe, 
             ncol = 2, strip.position = "right") +
  xlab("Amino Acid Depth") + 
  xlim(-0.5,16) +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(colour = "black"),
        strip.text.y.right = element_text(angle = 0),
        legend.position = "none") + 
  ylab("% of amino acids") + 
  scale_y_continuous(expand = expansion(mult = c(0,0.05)))

```

``` {R Import Weerapana 10uM vs 100uM or 10uM vs 10uM}

we.conc.raw <- dir_ls("weerapana_2010_raw", regexp = "raw\\/Wee.*Concentrations") %>%
  readxl::read_excel(col_names = F) %>%
  select(2,5,7:9) %>%
  rename(protein_id = "...2", 
         Stripped.Sequence = "...5",
         ratio.1.1 = "...7",
         ratio.10.1 = "...8",
         Run = "...9") 

we.conc.match <- we.conc.raw %>%
  filter(!is.na(protein_id)) %>%
  select(protein_id, Stripped.Sequence)

we.conc.df <- we.conc.raw %>%
  mutate(Stripped.Sequence = sub(Stripped.Sequence,
                                  pattern = "^[RK]{1}\\.{1}(.*)\\.{1}[[:upper:]]{1}",
                                  replacement = "\\1"),
         Stripped.Sequence = gsub(Stripped.Sequence,
                                  pattern = "\\*",
                                  replacement = "")) %>% ## format all peptide sequences
  select(-protein_id) %>%
  left_join(we.conc.match, by = "Stripped.Sequence") %>%
  na.omit() %>%
  filter(Run %in% 1:6) %>%
  group_by(protein_id, Stripped.Sequence) %>%
  summarise(Reactivity = mean(ratio.10.1),
            .groups = "drop") %>%
  mutate(reactivity_bin = cut(Reactivity, c(0,2,15)),
         reactivity_bin = sub(reactivity_bin, 
                              pattern = "\\((.*)\\,(.*)\\]",
                              replacement = "\\1-\\2"),
         protein_id = sub(protein_id, 
                          pattern = ".*\\: (.*)",
                          replacement = "\\1"),
         CysType = ifelse(reactivity_bin == "0-2",
                          yes = "Significantly labelled at 10uM",
                          no = "Significantly labelled at 100uM")) %>%
  left_join(human.fasta) %>%
  mutate(Peptide.Cys = str_locate(Stripped.Sequence, ## Location of Cys in peptide
                                  "C")[,1] - 2, # account for +1 of matching
         Peptide.Start = str_locate(string = Sequence, ## find peptide location in protein
                                    pattern = Stripped.Sequence)[,1], ## [1] takes start of seq
         position = Peptide.Start + Peptide.Cys + 1) %>%
  left_join(af.pred, by = c("protein_id", "position")) %>%
  filter(!is.na(HalfShell)) 

we.conc.df %>%
  rbind(we.conc.df %>% mutate(CysType = "All")) %>%
  mutate(Total = length(AA)) %>%
  group_by(AA, CysType, reactivity_bin, Total, HalfShell) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Part = n / Total) %>%
  distinct() %>%
  arrange(CysType) %>% 
  ggplot(aes(x = HalfShell,
            # y = log.intensity,
             y = Part)) + 
  geom_col() +
  #geom_density_2d_filled(n=50) +
  facet_wrap(~CysType, 
             ncol = 1,
             scales = "free_y") +
  xlab("Amino Acid Depth") + 
  xlim(-0.5,16) +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(colour = "black", size = 12, face = "bold"),
        #strip.text.y.right = element_text(angle = 0),
        legend.position = "none") + 
  ylab("% of amino acids") + 
  scale_y_continuous(expand = expansion(mult = c(0,0.05)))

```

``` {R Import Weerapana 10uM vs 100uM or 10uM vs 10uM}

we.time.raw <- dir_ls("weerapana_2010_raw", regexp = "raw\\/Wee.*Time") %>%
  readxl::read_excel(col_names = F) %>%
  select(2,5,7:8) %>%
  rename(protein_id = "...2", 
         Stripped.Sequence = "...5",
         ratio.10.1 = "...7",
         Run = "...8") 

we.time.match <- we.time.raw %>%
  filter(!is.na(protein_id)) %>%
  select(protein_id, Stripped.Sequence)

we.time.df <- we.time.raw %>%
  mutate(Stripped.Sequence = sub(Stripped.Sequence,
                                  pattern = "^[RK]{1}\\.{1}(.*)\\.{1}[[:upper:]]{1}",
                                  replacement = "\\1"),
         Stripped.Sequence = gsub(Stripped.Sequence,
                                  pattern = "\\*",
                                  replacement = "")) %>% ## format all peptide sequences
  select(-protein_id) %>%
  left_join(we.time.match, by = "Stripped.Sequence") %>%
  na.omit() %>%
  filter(Run %in% 1:6) %>%
  group_by(protein_id, Stripped.Sequence) %>%
  summarise(Reactivity = mean(ratio.10.1),
            .groups = "drop") %>%
  mutate(reactivity_bin = cut(Reactivity, c(0,2.5,15)),
         reactivity_bin = sub(reactivity_bin, 
                              pattern = "\\((.*)\\,(.*)\\]",
                              replacement = "\\1-\\2"),
         protein_id = sub(protein_id, 
                          pattern = ".*\\: (.*)",
                          replacement = "\\1"),
         CysType = ifelse(reactivity_bin == "0-2.5",
                          yes = "Significantly labelled at 6 minutes",
                          no = "Significantly labelled at 60 minutes")) %>%
  left_join(human.fasta) %>%
  mutate(Peptide.Cys = str_locate(Stripped.Sequence, ## Location of Cys in peptide
                                  "C")[,1] - 2, # account for +1 of matching
         Peptide.Start = str_locate(string = Sequence, ## find peptide location in protein
                                    pattern = Stripped.Sequence)[,1], ## [1] takes start of seq
         position = Peptide.Start + Peptide.Cys + 1) %>%
  left_join(af.pred, by = c("protein_id", "position")) %>%
  filter(!is.na(HalfShell)) 

we.time.df %>%
  rbind(we.time.df %>% mutate(CysType = "All")) %>%
  mutate(Total = length(AA)) %>%
  group_by(AA, CysType, reactivity_bin, Total, HalfShell) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Part = n / Total) %>%
  distinct() %>%
  arrange(CysType) %>% 
  ggplot(aes(x = HalfShell,
            # y = log.intensity,
             y = 100*Part)) + 
  geom_col() +
  #geom_density_2d_filled(n=50) +
  facet_wrap(~CysType, 
             ncol = 1,
             scales = "free_y") +
  xlab("Amino Acid Depth") + 
  xlim(-0.5,16) +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(colour = "black", size = 12, face = "bold"),
        #strip.text.y.right = element_text(angle = 0),
        legend.position = "none") + 
  ylab("% of amino acids") + 
  scale_y_continuous(expand = expansion(mult = c(0,0.05)))


```

``` {R OxiMouse -- use cytosol and secreted as two diff compositions of oxidised cysteines}

af.mus <- dir_ls("/Users/mew21/Library/CloudStorage/OneDrive-ImperialCollegeLondon/CysteineAccessibility/musmus/output", regexp = "AlphaFold") %>%
  fread() %>%
  rename(HalfShell = "nAA_12_70_pae",
         WholeShell = "nAA_24_180_pae")

mus.cys.wp <- af.mus %>%
  filter(AA == "C") %>%
  select(protein_id, AA, position, HalfShell) %>%
  mutate(Total = length(AA),
         Class = "Whole Proteome") %>%
  group_by(AA, #log.intensity, 
           Total, HalfShell) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Part = n / Total)

oxi.raw <- dir_ls("oximouse", regexp = "supp") %>%
  readxl::read_excel(sheet = "Single sites high confi. young") %>%
  row_to_names(row_number = 1)

oxi.df <- oxi.raw %>%
  select(1:15) %>%
  pivot_longer(cols = 6:15,
               names_to = "tissue",
               values_to = "perc_ox") %>%
  filter(perc_ox != "NA") %>%
  mutate(perc_ox = sub(perc_ox,
                       pattern = ".*",
                       replacement = ""),
         perc_ox = as.numeric(perc_ox),
         Site = as.integer(Site)) %>%
  rename(position = "Site",
         protein_id = "Uniprot ID") %>%
  left_join(af.mus)
  

## Number of annotated (AF) sites
oxi.df %>%
  filter(!is.na(HalfShell)) %>%
  nrow()

oxi.df %>%
  filter(is.na(HalfShell)) %>%
  nrow()
  
oxi.df %>%
  pull(Motif) %>%
  unique() %>%
  length()

##


## Number of cysteines annotated for each tissue type
oxi.df %>%
  select(protein_id, position, tissue) %>%
  distinct() %>%
  group_by(tissue) %>%
  summarise(n=n())

oxi.protein %>%
  select(`Uniprot ID`, tissue) %>%
  distinct() %>%
  group_by(tissue) %>%
  summarise(n=n())

oxi.df %>%
  ggplot(aes(x = perc_ox)) +
  geom_density() + 
  facet_wrap(~tissue, ncol =1)


```

``` {R Mus musculus fasta reformat}

index <- round(seq(1,length(mus.fasta.raw), length.out = 8),0)

mus.fasta <- data.frame()

for(j in 1:7){

  tmp.df <- lapply(index[j]:(index[j+1]-1), 
                          function(i) rbind(data.frame(Gene = names(mus.fasta.raw)[[i]], 
                                                       Sequence = paste(mus.fasta.raw[[i]], collapse = "")
                                                       )
                                            )
                          ) %>% 
  bind_rows()
  
  mus.fasta <- rbind(mus.fasta, tmp.df)
  
  }

mus.fasta <- mus.fasta %>%
  mutate(protein_id = sub(Gene, 
                          pattern = "..\\|(.*)\\|.*",
                          replacement = "\\1"))

rm(mus.fasta.raw)
rm(tmp.df)

```

``` {R Plot some distributions based on % oxidation and tissue type}

## Average from whole mouse proteome as reference:
mouse.proteome.ref <- af.mus %>%
  filter(AA == "C") %>%
  mutate(Total = length(AA)) %>%
  group_by(Total, HalfShell) %>%
  summarise(n=n(), .groups = "drop") %>%
  mutate(AA = "C",
         Part = n / Total,
         ox_bin_name = "Cysteines - whole proteome")
  
oxi.all.tissue.dist %>%
  filter(AA == "C") %>%
  nrow()

oxi.all.tissue.dist <- oxi.df %>%
  mutate(ox_bin = cut(perc_ox, c(0,10,20,100)),
         ox_bin = sub(ox_bin, 
                      pattern = "\\((.*)\\,(.*)\\]",
                      replacement = "\\1-\\2")) %>%
  left_join(data.frame(ox_bin = c("0-10", "10-20", "20-100"),
                       ox_bin_name = c("0-10% oxidised", "10-20% oxidised", "20%+ oxidised"))) %>%
  left_join(mus.fasta) %>%
  mutate(Peptide.Cys = str_locate(Motif, ## Location of Cys in peptide
                                  "C")[,1] - 2, # account for +1 of matching
         Peptide.Start = str_locate(string = Sequence, ## find peptide location in protein
                                    pattern = Motif)[,1], ## [1] takes start of seq
         position = Peptide.Start + Peptide.Cys + 1) %>%
  filter(!is.na(HalfShell),
         AA == "C") %>%
  na.omit() %>%
  distinct() %>%
  group_by(ox_bin_name, tissue) %>%
  mutate(Total = n()) %>%
  ungroup() %>% 
  group_by(AA, ox_bin_name, Total, HalfShell, tissue) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Part = n / Total) %>%
  distinct() 


oxi.all.tissue.dist %>%
  #rbind(mouse.proteome.ref %>% select(colnames(oxi.all.tissue.dist))) %>%
  ggplot(aes(x = HalfShell,
            # y = log.intensity,
             y = 100*Part)) + 
  geom_col() +
  #geom_density_2d_filled(n=50) +
  facet_grid(ox_bin_name~tissue,#, "Cysteines - whole proteome")), 
             #ncol = 1,
             scales = "free_y") +
  xlab("Amino Acid Depth") + 
  xlim(-0.5,21) +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(colour = "black", size = 12, face = "bold"),
        #strip.text.y.right = element_text(angle = 0),
        legend.position = "none") + 
  ylab("% of amino acids") + 
  scale_y_continuous(expand = expansion(mult = c(0,0.05)))


```

``` {R Compare protein abundance to cysteine availability}

### Use lung for comparison

oxi.protein.raw <- dir_ls("oximouse", regexp = "supp") %>%
  readxl::read_excel(sheet = "Protein abundance") %>%
  row_to_names(row_number = 1)

oxi.protein <- oxi.protein.raw %>%
  select(c(1,3:12)) %>%
  pivot_longer(cols = 2:11,
               names_to = "tissue",
               values_to = "abundance") %>%
  filter(abundance != "NA") %>%
  mutate(abundance = sub(abundance,
                     pattern = ".*",
                     replacement = ""),
         abundance = as.numeric(abundance))

oxi.cys.prot <- oxi.df %>%
  left_join(oxi.protein %>% 
              rename(protein_id = "Uniprot ID") %>%
              mutate(tissue = ifelse(tissue == "Lung young",
                                     yes = "Lung",
                                     no = "Else")),
            by = c("protein_id", "tissue")) %>%
  filter(!is.na(abundance),
         !is.na(HalfShell))  

oxi.cys.prot %>%
  mutate(abundance_bin = as.character(cut(log2(abundance), 20)),
         abundance_bin = as.numeric(sub(abundance_bin, 
                              pattern = "\\((.*)\\,(.*)\\]",
                              replacement = "\\1"))) %>%
  group_by(HalfShell, abundance_bin) %>%
  summarise(n=n()) %>% 
  ggplot(aes(x = HalfShell,
             y = abundance_bin,
             fill = n)) +
  geom_tile() +
  scale_fill_viridis() + 
  #geom_density_2d_filled(n=50) +
  xlab("Amino Acid Depth") + 
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(colour = "black", size = 12, face = "bold"),
        #strip.text.y.right = element_text(angle = 0),
        legend.position = "none") + 
  ylab("Protein abundance") + 
  scale_y_continuous(expand = expansion(mult = c(0,0))) +
  scale_x_continuous(expand = expansion(mult = c(0,0)))

  

```

``` {R Phosphorylation dependent cys reactivity}

## Try to get whole proteome analysis of asynch. cells to use as proteome abundance
ek.wp.asynch <- dir_ls("kemper_phos/", regexp = "386") %>%
  fread() %>%
  filter(V1 == "P" | str_detect(V2, "PLINE")) %>% 
  mutate(V1 = ifelse(V1 == "H",
                     yes = NA,
                     no = V1)) %>%
  shift_row_values(at = 1) %>%
  row_to_names(1) %>%
  #clean_names() %>%
  .[,-43] %>%
  .[,c(2:14, 20:24, 30:34)] %>% ## Channel 1,3,5 are asynch replicates
  pivot_longer(cols = 9:23,
               names_to = "Channel",
               values_to = "Quantity") %>%
  separate(Channel, into = c("Measurement", "Channel"), sep = " m/z_") %>%
  rename(protein_id = "LOCUS") %>%
  filter(Measurement == "norm_total",
         PEP_NUM > 1) %>% ## use norm_total for quantification
  mutate(Channel = round(as.numeric(Channel,0))) %>%
  group_by(protein_id) %>%
  summarise(Mean.Protein.Quantity = mean(as.numeric(Quantity)),
            SD.Protein.Quantity = sd(as.numeric(Quantity)),
            CV.Protein = 100 * SD.Protein.Quantity / Mean.Protein.Quantity,
            .groups = "drop")
     
## check distributions of norm_total signal -- looks good normlaisation          
ek.wp.asynch %>%
  filter(Measurement == "norm_total") %>%
  ggplot(aes(x = log2(as.numeric(Quantity)),
             fill = Channel,
             group = Channel,
             alpha = 0.5)) + 
  geom_density()

## Read in cysteine reactivity data (asynch only)
ek.cys.asynch <- dir_ls("kemper_phos/", regexp = "420") %>%
  fread() %>%
  filter(V1 != "H" | str_detect(V2, "SLINE")) %>% 
  mutate(V1 = ifelse(V1 == "H",
                     yes = NA,
                     no = V1)) %>%
  shift_row_values(at = 1) %>%
  row_to_names(1) %>%
  select(1:3, which(str_detect(colnames(.), "norm.*128") == T), 25:31) %>% ## Channel 4,5 are asynch replicates
  mutate(UNIQUE = ifelse(UNIQUE == "U",
                        yes = NA,
                        no = UNIQUE)) %>%
  fill(UNIQUE) %>%
  rename(protein_id = "UNIQUE") %>%
  filter(str_detect(SEQUENCE, "[[:upper:]]"),
         SLINE != "P") %>%
  pivot_longer(cols = 4:5,
               names_to = "Channel",
               values_to = "Quantity") %>%
  mutate(Channel = ifelse(str_detect(Channel,"128.128116"),
                          yes=1,
                          no=2),
         Quantity = as.numeric(Quantity)) %>%
  filter(Quantity != 0,
         !str_detect(protein_id, "Reverse"),
         #`Signal-noise` > 5,
         protein_id != "") %>%
  group_by(ScanNum, Filename) %>%
  filter(n() == 2) %>%
  ungroup() %>%
  group_by(protein_id, Filename, SEQUENCE, ScanNum) %>%
  summarise(Mean.Peptide.Quantity = mean(as.numeric(Quantity)),
            SD.Peptide.Quantity = sd(as.numeric(Quantity)),
            CV.Peptide = 100 * SD.Peptide.Quantity / Mean.Peptide.Quantity,
            .groups = "drop") %>%
  rename(Sequence = "SEQUENCE")


ek.cys.asynch %>%
  ggplot(aes(x = as.numeric(`Signal-noise`))) +
  geom_density() + 
  xlim(0,40)

ek.cys.asynch$Sequence %>% unique() %>% length()

intersect(ek.cys.asynch$protein_id,
          ek.wp.asynch$protein_id) %>% length()

ek.cys.asynch %>%
  ggplot(aes(x = as.numeric(TMT_purity))) + 
  geom_density() 

## Exp 476 taken as exemplar cysteine reactivity - couldn't find processed data online

## Join protein abundance data (relative to sample median...) with peptide abundance

ek.compare <- ek.cys.asynch %>%
  left_join(ek.wp.asynch, by = "protein_id") %>%
  group_by(protein_id, Sequence) %>%
  #filter(n() < 2) %>%
  ungroup()

ek.compare %>%
  group_by(Sequence) %>%
  filter(row_number()==1) %>%
  ungroup() %>%
  mutate(N.Cys = str_count(Sequence, "C"),
         Peptide.Length = nchar(Sequence)) %>%
  ggplot(aes(x = log2(Mean.Peptide.Quantity),
             y = log2(Mean.Protein.Quantity),
             colour = as.factor(N.Cys))) +
  geom_point(alpha = 0.7)

### TODO -- why does distribution of peptide intensities look bimodal?
## Most likely due to filtering (S/N and N-peptides in grouping)

## Join with AF data and plot accessibility vs. peptide and protein abundance
ek.af <- ek.compare %>%
  rename(Peptide.Sequence = "Sequence") %>%
  mutate(Stripped.Sequence = sub(Peptide.Sequence, 
                                 pattern = "^[RK]\\.([[:upper:]]*)\\(398\\.2529\\)([[:upper:]]*)\\..*", 
                                 replacement = "\\1*\\2")) %>%
  left_join(human.fasta, by = "protein_id") %>%
  mutate(Peptide.Cys = str_locate(Stripped.Sequence, ## Location of Cys in peptide
                                  "C\\*")[,1] - 2,
         Peptide.Start = str_locate(string = Sequence, ## find peptide location in protein
                                    pattern = Stripped.Sequence)[,1], ## [1] takes start of seq
         position = Peptide.Start + Peptide.Cys + 1) %>%
  left_join(af.pred, by = c("protein_id", "position")) %>%
  filter(!is.na(HalfShell),
         !is.na(Mean.Protein.Quantity),
         !is.na(Mean.Peptide.Quantity)) 

### How to plot abundance vs accessibility data
# 2D heatmap (with extra plot on top of HalfShell dist - background proteome)
# Generalise this code? Not for first iteration - easy to do form there

## Steps:
# Bin protein and peptide (log) abundances (20 bins)
# Plot binned data in 2d with colouring for numbers
# Instead plot as heatmap normalised to:
  # Average distribution across experiment
  # Proteome average for Cys

ek.prot.2d.df <- ek.af %>%
  mutate(peptide.rank = as.numeric(cut_number(.$Mean.Peptide.Quantity, 10)),
         protein.rank = as.numeric(cut_number(.$Mean.Protein.Quantity, 10)),
         Total = length(AA)) 

ek.prot.2d.df %>%
  group_by(Total, protein.rank, HalfShell) %>%
  summarise(n=n(),
            .groups = "drop") %>%
  mutate(Part = 100 * n / Total,
         #Part.Norm = Part - Ref.Part
         ) %>% 
  ggplot(aes(x = HalfShell,
             y = protein.rank,
             fill = Part)) +
  geom_tile() + 
  scale_fill_gradient2() + 
  scale_y_continuous(limits = c(0.5,10.5),
                     expand = expansion(mult = c(0,0))) +
  scale_x_continuous(limits = c(-0.5,19.5),
                     expand = expansion(mult = c(0,0))) +
  ylab("Protein abundance bins") +
  theme(panel.grid = element_blank())


ek.prot.2d.df %>%
  group_by(Total, peptide.rank, HalfShell) %>%
  summarise(n=n(),
            .groups = "drop") %>%
  #left_join(human.cys.ref.dist, by = "HalfShell") %>%
  mutate(Part = 100 * n / Total,
         #Part.Norm = Part - Ref.Part
         ) %>% 
  ggplot(aes(x = HalfShell,
             y = peptide.rank,
             fill = Part)) +
  geom_tile() + 
  scale_fill_gradient2() + 
  scale_y_continuous(limits = c(0.5,10.5),
                     expand = expansion(mult = c(0,0))) +
  scale_x_continuous(limits = c(-0.5,19.5),
                     expand = expansion(mult = c(0,0))) +
  ylab("Peptide abundance bins") +
  theme(panel.grid = element_blank())

### TODO:
## Plot distributions minus WP distributions for EK data (1D)
## Import lists of reactivity/abundance responsive sites (mitosis/asynch)
## Pull out 'ligandable' sites from Cell paper
## Pull out sites from SLC-ABPP and bin by number of liganded (R > 5) to identify 'highly druggable sites' -- look for bias in structure and accessibility


```

``` {R Use mitosis/asynch reactivity changes for enrichment}

## Just classify as reactivity or abundanc echages (no polarity for now)
ek.mit.reactivity <- dir_ls("kemper_phos/", regexp = "41592_2022_1398_MOESM2_ESM") %>%
  readxl::read_excel(sheet = 3,
                     col_names = F) %>%
  filter(!str_detect(`...1`, "All")) %>%
  row_to_names(row_number = 1) %>%
  filter(str_detect(`Reactivity-based change?`, "REACTIVITY") | 
           str_detect(`Expression-based change?`, "EXPRESSION")) %>%
  select(c(1,5:7)) %>%
  pivot_longer(cols = 3:4,
               values_to = "Change",
               names_to = "Temp") %>%
  filter(!is.na(Change)) %>%
  separate(gene_res, into = c("gene", "position"), sep = "_") %>%
  mutate(position = as.integer(position)) %>%
  select(-Temp, -gene) %>%
  rename(protein_id = "accession") %>%
  left_join(af.pred, by = c('protein_id', 'position')) %>%
  group_by(Change) %>%
  mutate(Total = n()) %>%
  ungroup()

ek.mit.reactivity %>%
  group_by(Total, structure_group, Change, HalfShell) %>%
  summarise(n=n(),
            .groups = "drop") %>%
  mutate(Part = 100 * n / Total,
         #Part.Norm = Part - Ref.Part
         ) %>% 
  ggplot(aes(x = HalfShell,
             y = Part,
             fill = Change)) +
  geom_col() + 
  facet_grid(Change~structure_group) +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(colour = "black", hjust = -0.01),
        strip.text.y.right = element_text(angle = 0),
        legend.position = "none") + 
  ylab("% of residues") + 
  scale_y_continuous(expand = expansion(mult = c(0,0.05))) + 
  xlab("Amino Acid Depth")


```

``` {R Extra 2D plots}
  
ek.prot.2d.df %>%
  select(Peptide.Sequence, log.peptide.quantity, log.protein.quantity) %>%
  distinct() %>%
  ggplot(aes(x=log.peptide.quantity,
             y=log.protein.quantity)) + 
  geom_density_2d_filled(bins = 20)
  geom_point()

```


``` {R LiP-MS Data to assess surface selectivity}


```

``` {R Side quest -- N-terminal glycine accessibility by myristoylation status}

### Ask wouter or james for N-myr list


```