---
title: "Cysteine accessibility analysis"
output: html_document
date: "`r Sys.Date()`"
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = F,
                      eval = T, 
                      warning = F,
                      cache = T,
                      cache.lazy = F,
                      fig.path = paste0(getwd(),
                                        "/plots/",
                                        gsub(Sys.Date(), ## Make folder for date of running
                                             pattern = "-", 
                                             replacement = ""),
                                        "/"))

library("tidyverse")
library("magrittr")
library("data.table")
library("fs")
library("janitor")
library(PNWColors)
library("ggtext")
library(ComplexHeatmap)
library(ComplexUpset)
library(geomtextpath)
library(scales)
library(broom)


```

``` {R Set absolute file paths for data to be read in}

### Set absolute paths to data directories
path.to.data.folder <-  getwd() ## Set to wherever the directory of the unzipped folder of raw/resource/processed data 

### Set date root of reformatted datasets - depends if the reformatting scripts were used
root.date <- "20221125"


```

``` {R Read in human fasta, AlphaFold accessibility, UniProt annotations}


### Naming convention for variables:
# res.* = resource: dataframe used at various points and kept in raw format for reference

## Read in human fasta
res.fasta <- dir_ls(paste0(path.to.data.folder, "/resources"), regexp = "FASTA") %>%
  fread() %>%
  mutate(protein_id = sub(Gene, 
                          pattern = "..\\|(.*)\\|.*",
                          replacement = "\\1")) %>%
  select(protein_id, Sequence)

#### Human AlphaFold predictions
res.af <- dir_ls(paste0(path.to.data.folder, "/resources"), regexp = "AlphaFoldPredicted.*hsapiens") %>% 
  fread() %>%
  rename(pPSE = "nAA_12_70_pae")


#Uniprot PTM annotations
res.uniprot.annotations <- dir_ls(paste0(path.to.data.folder, "/resources"), regexp = "allann") %>%
  fread()


## Filtered for confident prediction (quality > 70)
res.cys.distribution.filtered <- res.af %>%
  filter(AA == "C",
         quality > 70,
         !is.na(pPSE)) %>%
  select(protein_id, quality, AA, position, pPSE, structure_group) %>%
  distinct() %>%
  mutate(Total = length(AA),
         Class = "Whole Proteome") %>%
  group_by(Total, pPSE) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Ref.Part = n / Total) %>%
  select(pPSE, Ref.Part)

```

#### Figure 2

``` {R Figure 2a}

## Background distribution of average for all AAs
all.aa.ref <- res.af %>%
  filter(quality > 70,
         !is.na(pPSE)) %>%
  distinct() %>%
  mutate(Total = nrow(.)) %>%
  count(Total, pPSE, 
        name = "N.Per.pPSE") %>%
  mutate(Ref.Part = N.Per.pPSE / Total) %>%
  select(pPSE, Ref.Part)

## Set =< 5 as exposed and get % exposed/buried for each AA to order plots
all.aa.percent.exposed <- res.af %>%
  filter(quality > 70) %>%
  mutate(Exposed = ifelse(pPSE > 4,
                          yes = "Non-exposed",
                          no = "Exposed")) %>%
  count(AA, Exposed) %>%
  pivot_wider(names_from = Exposed,
              values_from = n) %>%
  mutate(`% Exposed` = 100 * Exposed / (Exposed + `Non-exposed`))

## Distribution for each AA
all.aa.pPSE <- res.af %>%
  filter(quality > 70,
         !is.na(pPSE)) %>%
  distinct() %>%
  add_count(AA, name = "Total.Per.AA") %>%
  count(Total.Per.AA, pPSE, AA, 
        name = "N.Per.pPSE") %>%
  mutate(Part = N.Per.pPSE / Total.Per.AA) 


## As heatmap
all.aa.pPSE %>%
  left_join(all.aa.ref, by = c("pPSE")) %>%
  mutate(Part.Norm = Part - Ref.Part,
         AA = factor(AA, levels = c(all.aa.percent.exposed$AA[order(all.aa.percent.exposed$`% Exposed`)]))) %>% ## Order by % exposed residues
  #filter(quality > 70) %>%
  #select(AA, pPSE, Part.Norm, Total) %>%
  ggplot(aes(x=pPSE,
             y=AA,
             fill=100*Part.Norm)) + 
  geom_tile() +
  scale_fill_gradientn(breaks = c(-10, 0, 10, 20),
                       minor_breaks = NULL,
                       colours = c("#FF5C5C", "white", "#6666FF", "#0000DC"),
                       limits = c(-11.5,23)) + 
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        legend.position = "top",
        panel.background = element_rect(fill = "#F6F6F6"),
        axis.text.y = element_text(size = 12.5)) + 
  scale_x_continuous(expand = expansion(mult = c(0.0,0.0)),
                     limits = c(-0.5,16.5)) + 
  scale_y_discrete(expand = expansion()) +
  labs(fill = "% enrichment vs.\nproteome average",
       y = "",
       x = "pPSE")

#ggsave("plots/20221027/Figure2a.png", height = 5, width = 6)

#rm(list = ls()[str_detect(ls(), "all\\.aa")])

```

``` {R Figure 2b}

uniprot.combined <- dir_ls(paste0(path.to.data.folder, "/formatted_data"), regexp = paste0(root.date, "_figure2_uniprot")) %>%
  fread()

uniprot.facets <- unique(uniprot.combined$Facet)

uniprot.facet.order <- c(uniprot.facets[str_detect(uniprot.facets, "Disulfide")],
                         uniprot.facets[str_detect(uniprot.facets, "Active")],
                         uniprot.facets[str_detect(uniprot.facets, "Palmitoyl")],
                         uniprot.facets[str_detect(uniprot.facets, "Prenyl")])
                         


uniprot.combined %>%
  mutate(Facet = factor(Facet, levels = uniprot.facet.order)) %>%
  filter(AA=="C") %>%
  group_by(Class) %>%
  mutate(Total = n()) %>% 
  ungroup() %>%
  group_by(AA,  Facet,Class, Total, pPSE) %>%
  summarise(n = n(), .groups = "drop") %>%
  mutate(Part = n / Total) %>%
  #left_join(human.cys.ref.dist, by = "pPSE") %>% 
  #mutate(Part.Norm = Part - Ref.Part) %>%
  ggplot(aes(x = pPSE,
             y = 100*Part,
             fill = Class)) + 
  geom_col(position = position_identity()) +
  facet_wrap(~Facet, 
             ncol = 2) + 
  xlab("pPSE") + 
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(colour = "black", hjust = -0.01, size = 11),
        legend.position = "none") + 
  ylab("% of residues") + 
  scale_y_continuous(expand = expansion(mult = c(0.0,0.05))) +
  scale_x_continuous(expand = expansion(mult = c(0.0,0.05)),
                     limits = c(-0.5, 14.5)) +
  scale_fill_manual(values = pnw_palette("Sunset2", 4))

#ggsave("plots/20221027/Figure2b.png", height = 4, width = 3.9)

#rm(list = ls()[str_detect(ls(), "$uniprot")])

```

``` {R Figure 2c}

### Heat-denatured vs. native lysates in 2-channel isoTOP-ABPP (from Backus et al., Nature, 2016)
heat.denature.raw <- dir_ls(paste0(path.to.data.folder, "/data/backus_nature_2016"), regexp = "/41586.*xlsx") %>%
  readxl::read_excel(sheet = "heat_inact_IA_alkyne") %>%
  select(1,3) %>%
  separate(1, into = c("protein_id", "position"), sep = "_") %>%
  rename(R = 3) %>% ## R = Active/Inactive
  filter(!str_detect(position, ",")) %>% ## Check no multiply annotated cys
  mutate(position = as.integer(sub(position, pattern = "C", replacement = ""))) %>%
  left_join(res.af, by = c("protein_id", "position")) %>%
  distinct() %>%
  filter(!is.na(pPSE),
         quality > 70,
         AA == "C") %>%
  mutate(R.bin = ifelse(R < 0.5,
                        yes = "Denaturation-enriched",
                        no = ifelse(R < 2, 
                                    yes = "Non-changing",
                                    no = "Native-enriched"))) 


heat.denature.enriched <- heat.denature.raw %>%
  add_count(R.bin, name = "Total") %>%
  count(pPSE, Total, R.bin) %>%
  mutate(Part = n / Total,
         facet = paste0(R.bin, "\nn = ",Total))
  

### All detected sites in heat denature experiment
heat.denature.ref <- heat.denature.raw %>%
  add_count(name = "Total") %>%
  count(pPSE, Total) %>%
  mutate(Part = n / Total)

## R < 0.5 --> preferentially reactive in denatured lysate
## R > 2 --> preferentially reactive in native lysate

heat.denature.enriched %>%
  filter(R.bin != "Non-changing") %>% ## Plot only native- or heat-enriched cysteines
  mutate(facet = factor(facet, levels = unique(heat.denature.enriched$facet)[c(2,3)])) %>% 
  ggplot() +
  #geom_col(inherits.aes = F,
  #        data = heat.denature.ref,
  #         aes(x = pPSE,
  #             y = 100*Part),
  #         position = position_identity(),
  #         fill = "light grey",
  #         alpha = 0.5) +
  geom_col(aes(x = pPSE,
               y = 100*Part,
               fill = R.bin),
           position = position_identity(),
           alpha = 0.45) +
  theme_linedraw() + 
  #facet_wrap(~facet, ncol = 2) +
  theme(panel.grid = element_blank(),
        legend.position = "bottom",
        legend.text=element_text(size=14),
        axis.title = element_text(size=16),
        strip.background = element_rect(fill = "white", colour = "white"),
        strip.text = element_text(colour = "black", hjust = -0.01, size = 12)) + 
  ylab("% of residues") + 
  xlab("pPSE") +
  scale_x_continuous(expand = expansion(mult = c(0.01,0.01)),
                     limits = c(-0.55, 15.55)) +
  scale_y_continuous(expand = expansion(mult = c(0.0, 0.05))) + 
  scale_fill_manual(values = c(pnw_palette("Bay", 8)[8], 
                               pnw_palette("Shuksan2", 5)[1])) +
  guides(fill=guide_legend(ncol=1)) +
  labs(fill = "")


#ggsave("plots/20221027/Figure2c.png", height = 3.33, width = 3)

#rm(list = ls()[str_detect(ls(), "heat.denature")])


``` 

``` {R Figure 2e}
  
cp.coverage.raw <- dir_ls(paste0(path.to.data.folder, "/formatted_data"), regexp = paste0(root.date, "_figure2_sites")) %>%
  fread()

cp.coverage <- cp.coverage.raw %>%
  filter(Reference != "Weerapana et al.", ## Remove original isoTOP -- use Backus 2016 for overview of isoTOP
         Reference != "Vinogradova et al.") %>% ## Remove Vinogradova (other IA-DTB - Kemper et al. - shows same trend)
  add_count(Dataset, Reference, Labelling, Probe, name = "Total") %>%
  mutate(facet = paste0(Dataset, "\n", 
                        Reference,", ", 
                        Year, "\n", 
                        Labelling, ", ",
                        sub(Probe, pattern = ".*\\ (.*)", replacement = "\\1"))) %>%
  count(pPSE, Total, Probe, facet) %>%
  mutate(Part = n / Total)


cp.facet <- unique(cp.coverage$facet)

cp.facet.order <- c(cp.facet[str_detect(cp.facet, "Yan et al")],
                    cp.facet[str_detect(cp.facet, "DIA-ABPP")],
                    cp.facet[str_detect(cp.facet, "SLC")],
                    cp.facet[str_detect(cp.facet, "Kemper")],
                   # cp.facet[str_detect(cp.facet, "Vino")],
                    #cp.facet[str_detect(cp.facet, "Nature, 2010")],
                    cp.facet[str_detect(cp.facet, "mCSCP")],#
                    cp.facet[str_detect(cp.facet, "Benziodoxole")],
                    cp.facet[str_detect(cp.facet, "UK")],
                    cp.facet[str_detect(cp.facet, "Backus et")],
                    cp.facet[str_detect(cp.facet, "N-acryl")],
                    cp.facet[str_detect(cp.facet, "Bromoketone")], 
                    cp.facet[str_detect(cp.facet, "Caged-iodoketone")],
                    cp.facet[str_detect(cp.facet, "Caged-bromoketone")])


## Plot altogether
cp.coverage %>%
  mutate(facet = factor(facet, levels = cp.facet.order)) %>%
  ggplot() + 
  geom_col(data = res.cys.distribution.filtered,
           aes(x = pPSE,
               y = 100*Ref.Part),
           position = position_identity(),
           fill = "light grey",
           alpha = 0.6) +
  geom_col(aes(x = pPSE,
               y = 100*Part),
               fill = "forest green",
           position = position_identity(),
           alpha = 0.4) + 
  geom_text(data = cp.coverage %>% 
              select(facet, Total) %>% 
              distinct() %>% 
              mutate(facet = factor(facet, levels = cp.facet.order)),
            aes(x = 14.5,
                y = 17,
                hjust = 1,
                label = paste("n = ",Total)),
            size = 5) +
  facet_wrap(~facet, scales = "free_x", ncol = 6) +
  xlab("pPSE") + 
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        axis.title = element_text(size = 16),
        strip.background = element_blank(),
        strip.text = element_text(colour = "black", hjust = -0.01, size = 16),
        strip.text.y.right = element_text(angle = 0),
        legend.position = "none") + 
  ylab("% of amino acids") + 
  #scale_fill_manual(values = pnw_palette("Starfish", 3)) +
  scale_y_continuous(expand = expansion(mult = c(0.0,0.05))) +
  scale_x_continuous(limits = c(-0.55,14.5),
                     expand = expansion(mult = c(0.0,0.05)))

#ggsave("plots/20221027/Figure2e.png", height = 6.5, width = 16.5)

  
```


``` {R Figure 2d}

## Secondary structure names
secondary.structure.name <- data.frame(
  structure_group = sort(unique(cp.coverage.raw$structure_group)), ## Alphabetical order
  secondary.structure = c("Bend", "Helix", "Strand", "Turn", "Unstructured"))


### Bias in secondary structures comparing DBIA (SLC-ABPP, Kuljanin et al.) with caged-iodoketone (CIK4, ChemMedChem, 2016)
secondary.structure.df <- cp.coverage.raw %>%
  filter(str_detect(Probe, "DBIA") | str_detect(Probe, "CIK4")) %>% 
  #select(protein_id, structure_group, AA, position, pPSE, quality) %>%
  distinct() %>%
  filter(!is.na(pPSE),
         AA == "C",
         quality > 70) %>% 
  add_count(Probe, name = "Total") %>%
  mutate(facet = paste0(Labelling, "\n", Probe, "\nn = ", Total)) %>%
  left_join(secondary.structure.name) %>%
  select(-structure_group)

## Average for each secondary structure proteome-wide
secondary.structure.ref <- res.af %>%
  filter(AA == "C", 
         quality > 70,
         !is.na(pPSE)) %>%
  add_count(structure_group, name = "Total") %>%
  count(Total, structure_group, pPSE) %>%
  mutate(Ref.Part = n / Total) %>%
  left_join(secondary.structure.name) %>%
  select(-structure_group)

secondary.structure.df %>%
  add_count(facet, secondary.structure, name = "Total") %>%
  count(secondary.structure, Probe, 
           Total, pPSE) %>%
  mutate(Part = n / Total) %>%
  ggplot() + 
  #geom_col(data = secondary.structure.ref,
  #         aes(x = pPSE,
  #             y = 100*Ref.Part),
  #         position = position_identity(),
  #         fill = "grey",
  #         alpha = 0.25) +
  geom_col(aes(x = pPSE,
               y = 100*Part,
               fill = Probe),
           position = position_identity(),
           alpha = 0.35) +
  facet_grid(~secondary.structure) +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        strip.background = element_blank(),
        legend.position = "bottom",
        #legend.justification = "right",
        legend.text = element_text(size=11.5),
        strip.text = element_text(colour = "black", hjust = -0.01, vjust = 0, size = 12),
        strip.text.y.right = element_text(angle = 0)) + 
  ylab("% of residues") + 
  scale_y_continuous(expand = expansion(mult = c(0,0.05))) + 
  scale_x_continuous(limits = c(-0.55, 15.55), 
                     expand = expansion(mult = c(0,0.05))) + 
  guides(fill=guide_legend(ncol=2)) +
  labs(fill = "") +
  xlab("pPSE") + 
  scale_fill_manual(values = c(pnw_palette("Winter", 5)[4], pnw_palette("Sunset2", 5)[2]))

#ggsave("plots/20221027/Figure2d.png", height = 3, width = 7.5)

#rm(list = ls()[str_detect(ls(), "secondary")])


```




#### Figure 3

``` {R Figure 3a}

treatment.df <- dir_ls(paste0(path.to.data.folder, "/formatted_data"), regexp = paste0(root.date, "_figure3_treatment")) %>%
  fread()
  
## Plot together by faceting
treatment.df %>% 
  filter(!str_detect(Treatment, "Not liganded")) %>%
  mutate(Treatment = factor(Treatment, levels = c("Liganded in situ",
                                                  "Liganded in vitro",
                                                  "Not liganded in vitro"))) %>% 
  add_count(Treatment, name = "Total") %>% 
  count(pPSE, Total, Treatment) %>%
  mutate(Part = n / Total) %>% 
  ggplot() +
  geom_col(aes(x = pPSE,
               y = 100*Part,
               fill = Treatment),
           position = position_identity(),
           alpha = 0.25) +
  geom_col(data = treatment.df %>% 
             filter(str_detect(Treatment, "Not liganded")) %>%
             add_count(Treatment, name = "Total") %>% 
             count(pPSE, Total, Treatment) %>%
             mutate(Part = n / Total) %>%
             select(-Treatment),
           aes(x = pPSE,
               y = 100*Part),
               fill = "grey",
           position = position_identity(),
           alpha = 0.25) +
  geom_text(aes(x = 15, y = 17, 
                label = paste0("n = ", Total)),
            check_overlap = TRUE,
            hjust = 1) +
  theme_linedraw() + 
  facet_wrap(~Treatment) + 
  theme(panel.grid = element_blank(),
        legend.position = "none",
        strip.background = element_rect(fill = "white", colour = "white"),
        strip.text = element_text(colour = "black", hjust = -0.01, size = 12, face = "bold")) + 
  ylab("% of residues") + 
  xlab("pPSE") +
  scale_x_continuous(limits = c(-0.55, 15.55),
                     expand = expansion(mult = c(0.01,0.01))) +
  scale_y_continuous(expand = expansion(mult = c(0.0, 0.025)),
                     limits = c(0, 18)) + 
  scale_fill_manual(values = c(pnw_palette("Lake", 8)[4], 
                               pnw_palette("Shuksan2", 5)[1],
                               pnw_palette("Moth", 5)[1]))


#ggsave("plots/20221027/Figure3a.png", height = 3, width = 4.25)

#rm(list = ls()[str_detect(ls(), "fragment")])


```

``` {R Figure 3b}

fragment.treatment.overlap <- treatment.df %>%
  filter(Liganded == 1) %>%
  select(protein_id, position, Treatment, Liganded) %>%
  pivot_wider(names_from = "Treatment",
              values_from = "Liganded") %>%
  mutate_at(vars(3:4), replace_na, 0) %>%
  mutate(prot_pos = paste(protein_id, position, sep = "_")) %>%
  select(-protein_id, -position) %>%
  column_to_rownames(var = "prot_pos") %>%
  as.matrix() %>%
  make_comb_mat()


#fragment.treatment.overlap

```

``` {R Figure 3d}

figure.3d.df <- rbind(
  ### Fragment-liganded cysteines
  dir_ls(paste0(path.to.data.folder, "/formatted_data"), regexp = paste0(root.date, "_figure3_fragment_max-R")) %>%
    fread() %>%
    filter(R > 4) %>% ## Liganded in fragment lib
    mutate(Subset = "Liganded in\nfragment library") %>%
    select(protein_id,position,Dataset,quality,structure_group,AA,pPSE, Subset) %>%
    distinct(),
  
  ### Scout-liganded cysteines       
  dir_ls(paste0(path.to.data.folder, "/formatted_data"), regexp = paste0(root.date, "_figure3_scout")) %>%
    fread() %>%
    filter(R > 4) %>% ## Liganded in fragment lib
    mutate(Subset = "Liganded by\nscout fragments") %>%
    select(protein_id,position,Dataset,quality,structure_group,AA,pPSE, Subset) %>%
    distinct(),
  
  ### Non-liganded in fragment screening
  dir_ls(paste0(path.to.data.folder, "/formatted_data"), regexp = paste0(root.date, "_figure3_fragment_max-R")) %>%
    fread() %>%
    filter(R < 4) %>% ## Not liganded in fragment lib
    mutate(Subset = "Not liganded\n(fragment library)") %>%
    select(protein_id,position,Dataset,quality,structure_group,AA,pPSE, Subset) %>%
    distinct()) %>%
  add_count(Dataset, Subset,name="Total") %>%
  count(Dataset, Subset, Total, pPSE) %>%
  mutate(Part = n/Total)
                   

### Order facets
facet.top <- unique(figure.3d.df$Dataset)

facet.top.order <- c(facet.top[str_detect(facet.top, "Backus")],
                     facet.top[str_detect(facet.top, "Vino")],
                     facet.top[str_detect(facet.top, "Kul")],
                     facet.top[str_detect(facet.top, "Yang")])

   
### Plot altogether
figure.3d.df %>%
  mutate(Dataset = factor(Dataset, levels = facet.top.order)) %>%
  ggplot() +
  geom_col(data = res.cys.distribution.filtered,
           aes(x = pPSE,
               y = 100*Ref.Part),
          width = 1,
          position = position_identity(),
           fill = "light grey",
           alpha = 0.6) +
  geom_col(aes(x = pPSE,
               y = 100*Part,
               fill = Subset),
           width = 1,
           position = position_identity(),
           alpha = 0.25) +
  geom_text(aes(x = 15, y = 22.5, 
                label = paste0("n = ", Total)),
            check_overlap = TRUE,
            hjust = 1) +
  theme_linedraw() + 
  facet_grid(Subset ~ Dataset) + 
  theme(panel.grid = element_blank(),
        legend.position = "none",
        strip.background = element_rect(fill = "white", colour = "white"),
        strip.text = element_text(colour = "black", hjust = -0.01, size = 12, face = "bold")) + 
  ylab("% of residues") + 
  xlab("pPSE") +
  scale_x_continuous(limits = c(-0.55,15.5),
                     expand = expansion(mult = c(0.01,0.01))) +
  scale_y_continuous(expand = expansion(mult = c(0.0, 0.025))) + 
  scale_fill_manual(values = pnw_palette("Sunset2", 4))

#ggsave("plots/20221027/Figure3d.png", height = 6, width = 10.5)

#rm(list = ls()[str_detect(ls(), "figure.3d")])

```

``` {R Figure 3c}

## Off-targets of KRas G12C inhibitor (https://doi.org/10.1158/2159-8290.CD-15-1105)
sites.offtarget.g12c <- dir_ls(paste0(path.to.data.folder, "/data/patricelli_krasg12c"), regexp = "/2159") %>%
  readxl::read_excel(sheet = 1) %>%
  filter(`log-2Fold change (30uM853/DMSO)` < -2) %>%
  select(`Protein ID`, Sequence) %>%
  rename(protein_id = 1,
         Peptide.Sequence = 2) %>%
  left_join(res.fasta) %>%
  mutate(Peptide.Cys = str_locate(Peptide.Sequence, ## Location of Cys in peptide
                                  "C")[,1] - 2, # account for +1 of matching
         Peptide.Start = str_locate(string = Sequence, ## find peptide location in protein
                                    pattern = Peptide.Sequence)[,1], ## [1] takes start of seq
         position = Peptide.Start + Peptide.Cys + 1) %>%
  left_join(res.af, by = c("protein_id", "position"))
  


## Off-targets from various published cysteine profiling of lead-like compounds
sites.offtarget.literature <- data.frame(protein_id = c("Q9P2K8", # EIF2AK4 -- ARS-1620 (Kuljanin et al., Nat.Biotech., 2021)
                                                       #"P50613", # CDK7 -- THZ1 (Kuljanin et al., Nat.Biotech., 2021)
                                                       "Q5T953", # IER5L -- THZ1 (Kuljanin et al., Nat.Biotech., 2021)
                                                       "Q66PJ3", # ARL6IP4 -- THZ1 (Kuljanin et al., Nat.Biotech., 2021)
                                                       #"Q06187", # BTK -- Ibrutinib (Kuljanin et al., Nat.Biotech., 2021)
                                                       "P51451", # BLK -- Ibrutinib (Kuljanin et al., Nat.Biotech., 2021)
                                                       "P23458", # JAK1 -- VVD-118313 (Kavanagh et al., Nat.Chem.Biol., 2022)
                                                       "P29597", # TYK2 -- VVD-118313 (Kavanagh et al., Nat.Chem.Biol., 2022)
                                                       "P30519", # HMOX2 -- VVD-118313 (Kavanagh et al., Nat.Chem.Biol., 2022) 
                                                       "Q8N755", # SLC66A3 -- VVD-118313, (Kavanagh et al., Nat.Chem.Biol., 2022)
                                                       "Q5TFE4", # NT5DC1 -- CO-1686 (Niessen et al., Nat.Chem.Biol., 2017)
                                                       "Q99536", # VAT1 C86 -- Compound 1 (Wijeratne et al., ACS Med.Chem.Lett., 2018)
                                                       "Q99536", # VAT1 C50 -- Compound 1 (Wijeratne et al., ACS Med.Chem.Lett., 2018)
                                                       "Q08257"), # CRYZ C45 -- Compound 1 (Wijeratne et al., ACS Med.Chem.Lett., 2018)

                                        position = c(1245, ## Cysteine residues taken from literature SLC/isoTOP-ABPP data
                                                     #312, ## Removed as it is intended targets of lead compound
                                                     159,
                                                     171,
                                                     #481, ## Removed as it is intended targets of lead compound
                                                     313,
                                                     817,
                                                     838,
                                                     282,
                                                     135,
                                                     119,
                                                     86, 
                                                     50,
                                                     45))

sites.off.target.df <- rbind(sites.offtarget.literature,
                             sites.offtarget.g12c %>% select(protein_id, position)) %>%
  left_join(res.af) %>%
  filter(!is.na(position),
         !is.na(pPSE)) %>%
  select(protein_id, position, pPSE) %>%
  mutate(Dataset = "Lead-like\noff-targets") %>%
  distinct()

### FDA-approved targets
sites.fda.targets <- dir_ls(paste0(path.to.data.folder, "/resources"), regexp = "fda") %>%
  fread() %>%
  filter(str_detect(Residue, "CYS")) %>%
  mutate(position = as.numeric(sub(Residue, pattern = "CYS-(\\d*)",
                        replacement = "\\1"))) %>%
  select(protein_id, position) %>%
  left_join(res.af) %>%
  filter(!is.na(position),
         !is.na(pPSE)) %>%
  select(protein_id, position, pPSE) %>%
  mutate(Dataset = "FDA-approved\nCys targets") %>%
  distinct()

### Liganded cysteines by elaborated electrophiles (Vinogradova et al., 2020)
sites.elaborated.electrophiles <- dir_ls(paste0(path.to.data.folder, "/formatted_data"), regexp = paste0(root.date, "_figure3_fragment-all")) %>%
  fread() %>%
  filter(str_detect(Dataset, "Vino"),
         Frag.Class == "Elaborated electrophile",
         R > 4) %>%
  select(protein_id, position, pPSE) %>%
  mutate(Dataset = "Elaborated\nelectrophile") %>%
  distinct()

### All cysteines detected by MS-ABPP
sites.all.msabpp <- dir_ls(paste0(path.to.data.folder, "/formatted_data"), regexp = paste0(root.date, ".*figure2_sites")) %>%
  fread() %>%
  select(protein_id, position, pPSE) %>%
  filter(!is.na(position),
         !is.na(pPSE)) %>%
  select(protein_id, position, pPSE) %>%
  mutate(Dataset = "All Cys\nby MS-ABPP") %>%
  distinct()

### All cysteines detectable (quality > 70)
sites.all.cys <- res.af %>%
  filter(AA == "C",
         quality > 70) %>%
  select(protein_id, position, pPSE) %>%
  filter(!is.na(position),
         !is.na(pPSE)) %>%
  select(protein_id, position, pPSE) %>%
  mutate(Dataset = "All confidently-\npredicted Cys") %>%
  distinct()

figure.2e.df <- rbind(sites.all.cys, 
                      sites.all.msabpp, 
                      sites.off.target.df,
                      sites.fda.targets,
                      sites.elaborated.electrophiles) %>%
  add_count(Dataset, name = "Total")

figure.2e.facet <- figure.2e.df %>%
  #mutate(Facet = paste0(Dataset, "\nn = ", Total)) %>%
  pull(Dataset) %>%
  unique()

figure.2e.facet.order <- c(figure.2e.facet[str_detect(figure.2e.facet, "confident")],
                           figure.2e.facet[str_detect(figure.2e.facet, "MS-ABPP")],
                           figure.2e.facet[str_detect(figure.2e.facet, "Elaborated")],
                           figure.2e.facet[str_detect(figure.2e.facet, "Lead")],
                           figure.2e.facet[str_detect(figure.2e.facet, "FDA")])

figure.2e.total <- figure.2e.df %>%
  mutate(Facet = paste0(Dataset, "\nn = ", Total),
         Facet = factor(Facet, levels = figure.2e.facet.order)) %>%
  select(Dataset, Total) %>%
  distinct()

### Plot as boxplot
figure.2e.df %>%
  mutate(#Facet = paste0(Dataset, "\nn = ", Total),
         #Facet = factor(Facet, levels = figure.2e.facet.order),
         Dataset = factor(Dataset, levels = figure.2e.facet.order)) %>%
  ggplot(aes(x = pPSE,
             y = Dataset)) +
  geom_boxplot(fill = "dark blue",
               alpha = 0.5,
               width = 0.5,
               key_glyph = "rect") +
  geom_text(data = figure.2e.total,
            aes(x = 17.5,
                y = Dataset,
                label = paste0("n = ",Total)),
            vjust = -1,
            size = 3.5) +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        legend.position = "top",
        #axis.title = element_text(size = 15),
        #axis.text = element_text(size = 14),
        strip.background = element_rect(fill = "white", colour = "white"),
        axis.text.y = element_text(colour = "black", size = 13)) + 
  ylab("") + 
  xlab("pPSE") +
  labs(fill = "Liganded by:") +
  guides(fill = guide_legend(nrow = 3)) +
  xlim(0,21)

#ggsave("plots/20221027/Figure3c.png", height = 3, width = 5)

#rm(list = ls()[str_detect(ls(), "sites")])


```

#### Figure 4

``` {R Figure 4}

### Make function for generating combinatorial dataset of:
# Cysteine coverage
# Selectivity
# Size of druggable 'cysteinome'

function.coverage <- function(Cysteines.Identified,
                              Selectivity){
  
  temp.df <- crossing(Cysteines.Identified, ## Table of all combinations
                      Selectivity) %>%
    mutate(Total.Cys = 204707, ## from SP3-FAIMS paper
           Perc.Coverage = (100 * Cysteines.Identified * Selectivity) / Total.Cys, ## 
           #Perc.Coverage = 100 * Theoretical.Coverage / Druggable.Cys,
           Perc.Coverage = round(Perc.Coverage,1)) %>%
    #filter(Perc.Coverage <= 100)
    mutate(Perc.Coverage = ifelse(Perc.Coverage > 100,
                                  100,
                                  Perc.Coverage))
  
  return(temp.df)
  
}


function.coverage(Cysteines.Identified = seq(100,110000,by=50),
                  Selectivity = seq(0.01,12, by = 0.01)) %>% 
  filter(Perc.Coverage %in% c(10,25,50,90)) %>% 
  mutate(Perc.Coverage = factor(paste0(Perc.Coverage,"%"), 
                                levels = c("90%", "50%", "25%", "10%"))) %>%
  ggplot(aes(x = Selectivity,
             y = Cysteines.Identified/1000, 
             colour = Perc.Coverage,
             group = Perc.Coverage,
             label = Perc.Coverage)) + 
  scale_y_continuous(label = label_number(suffix = "k"),
                     limits = c(0,100),
                     expand = expansion(mult = c(0.05,0))) +
  scale_x_continuous(expand = expansion(mult = c(0.05,0)),
                     breaks = seq(0,12,by=2),
                     limits = c(0,11)) + 
  geom_textline(size = 4,
                hjust = 0.75,
                key_glyph = "rect") + 
  theme_linedraw() + 
  theme(panel.grid = element_blank()) + 
  scale_colour_manual(values = c("#003A6B","#2C5D87","#5880A2","#83A3BE")) +
  labs(colour = "Theoretical coverage of\ndruggable cysteines") + 
  geom_vline(xintercept = 1,
             linetype = "dashed",
             colour = "grey") + 
  xlab("Selectivity for\ndruggable cysteines") +
  ylab("Cysteines quantified\nper experiment")
  
#ggsave("plots/20221027/Figure4.png", height = 4, width = 6.5)


```

#### Supplementary Figures


``` {R Figure S1a}

upset.sites <- dir_ls(paste0(path.to.data.folder, "/formatted_data"), regexp = paste0(root.date, ".*figure2_sites")) %>%
  fread() %>%
  filter(str_detect(Dataset, "ABPP")) %>%
  select(protein_id, position, 
        Dataset) %>%
  mutate(R.bin = 1) %>%
  distinct() %>%
  pivot_wider(names_from = "Dataset",
              values_from = "R.bin") %>%
  mutate_at(vars(3:length(.)), replace_na, 0) %>%
  mutate(prot_pos = paste(protein_id, position, sep = "_")) %>%
  select(-protein_id, -position) %>%
  column_to_rownames(var = "prot_pos") %>%
  as.matrix() %>%
  make_comb_mat()

#png(filename = "plots/20221027/FigureS1a_AllSites.png", height = 3, width = 5, units = "in", res = 300)
ComplexHeatmap::UpSet(upset.sites)
#dev.off()

```

``` {R Figure S1b}

upset.scout.all <- dir_ls(paste0(path.to.data.folder, "/formatted_data"), regexp = paste0(root.date, ".*scout")) %>%
  fread() %>%
  select(protein_id, position, 
        Dataset) %>%
  mutate(R.bin = 1,
         Dataset = sub(Dataset, pattern = ".*\n(.*)$",
                       replacement = "\\1")) %>%
  distinct() %>%
  pivot_wider(names_from = "Dataset",
              values_from = "R.bin") %>%
  mutate_at(vars(3:length(.)), replace_na, 0) %>%
  mutate(prot_pos = paste(protein_id, position, sep = "_")) %>%
  select(-protein_id, -position) %>%
  column_to_rownames(var = "prot_pos") %>%
  as.matrix() %>%
  make_comb_mat()

#png(filename = "plots/20221027/FigureS1b_ScoutSites.png", height = 3, width = 5, units = "in", res = 300)
ComplexHeatmap::UpSet(upset.scout.all)
#dev.off()

```

``` {R Figure S1c}

upset.scout.liganded <- dir_ls(paste0(path.to.data.folder, "/formatted_data"), regexp = paste0(root.date, ".*scout")) %>%
  fread() %>%
  filter(R > 4) %>%
 select(protein_id, position, 
        Dataset) %>%
  mutate(R.bin = 1,
         Dataset = sub(Dataset, pattern = ".*\n(.*)$",
                       replacement = "\\1")) %>%
  distinct() %>%
  pivot_wider(names_from = "Dataset",
              values_from = "R.bin") %>%
  mutate_at(vars(3:length(.)), replace_na, 0) %>%
  mutate(prot_pos = paste(protein_id, position, sep = "_")) %>%
  select(-protein_id, -position) %>%
  column_to_rownames(var = "prot_pos") %>%
  as.matrix() %>%
  make_comb_mat()

#png(filename = "plots/20221027/FigureS1c_LigandedSites.png", height = 3, width = 5, units = "in", res = 300)
ComplexHeatmap::UpSet(upset.scout.liganded)
#dev.off()

```

``` {R Figure S2b}

## Top liganded fragment class per site used
figure.3d.df <- dir_ls(paste0(path.to.data.folder, "/formatted_data"), regexp = paste0(root.date, "_figure3_fragment-all-")) %>%
  fread() %>%
  filter(Frag.Class %in% c("Chloroacetamide", "Acrylamide")) %>%
  mutate(Liganded = ifelse(R > 4,
                           yes = Frag.Class,
                           no = "Not liganded")) #%>%
  #group_by(protein_id, position, Dataset, Liganded) %>%
  #filter(R == max(R)) %>% ## For each site and fragment class, keep only 1 row
  #ungroup()

figure.3d.df %>%
  mutate(Liganded = factor(Liganded, levels = c("Not liganded", "Chloroacetamide", "Acrylamide"))) %>%
  ggplot(aes(x = pPSE,
             y = Dataset,
             fill = Liganded)) +
  geom_boxplot(alpha = 0.5,
               key_glyph = "rect") +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        legend.position = "top",
        strip.background = element_rect(fill = "white", colour = "white"),
        strip.text = element_text(colour = "black", hjust = -0.01, size = 12, face = "bold")) + 
  ylab("") + 
  xlab("pPSE") +
  labs(fill = "Liganded by:") +
  scale_fill_manual(values = pnw_palette("Lake", 8)[c(8,6,4)],
                    labels = function(x) str_wrap(x, width = 100)) + 
  guides(fill = guide_legend(nrow = 3)) +
  xlim(0,21)

#ggsave("plots/20221027/FigureS2a_ClAc_Accessibility.png", height = 4, width = 4)

#rm(list = ls()[str_detect(ls(), "all\\.aa")])


```

``` {R Figure S2a}

res.chemical.properties <- dir_ls(paste0(path.to.data.folder, "/formatted_data"), regexp = "SMILES.xlsx") %>%
  readxl::read_excel(sheet = "ChemicalProperties")

res.frag.cys.df <- dir_ls(paste0(path.to.data.folder, "/formatted_data"), regexp = paste0(root.date, "_figure3_fragment-all-")) %>%
  fread() %>%
  filter(!str_detect(Dataset, "Vino")) 

fragment.ligandability <- res.frag.cys.df %>%
  add_count(Fragment, name = "Total.Cys.Interactions") %>%
  filter(R > 4) %>%
  count(Fragment, Frag.Class, Total.Cys.Interactions, name = "Sites.Liganded") %>%
  mutate(Perc.Liganded = 100 * Sites.Liganded / Total.Cys.Interactions) %>%
  left_join(res.chemical.properties) %>%
  select(Fragment, Frag.Class, Sites.Liganded, 9:17) %>% 
  pivot_longer(cols = 4:12,
               names_to = "Property",
               values_to = "Value")

fragment.ligandability %>%
  filter(Frag.Class %in% c("Chloroacetamide", "Acrylamide"),
         Sites.Liganded != 0,
         Sites.Liganded != 689) %>% ## Remove extreme outlier for plotting -- F11 from Yang et al
  ggplot(aes(x = Sites.Liganded,
             y = Value,
             #colour = Frag.Class
             )) + 
  geom_point(alpha = 0.25) + 
  facet_wrap(~Property, scales = "free") +
  theme_linedraw() + 
  theme(panel.grid = element_blank(),
        #legend.position = "none",
        strip.background = element_rect(fill = "white", colour = "white"),
        strip.text = element_text(colour = "black")) + 
  #xlim(0,200) + 
  scale_colour_manual(values = pnw_palette("Sunset2", 2)) + 
  scale_y_continuous(expand = expansion(mult = c(0.1,0.1))) +
  xlab("Sites liganded") + 
  labs(colour = "Warhead")

#ggsave("plots/20221027/FigureS2b_ChemicalProperties.png", height = 5.5, width = 6)

```

#### Summary statistics

``` {R Numbers for abstract}

### Unique cysteines in coverage dataframe
cp.coverage.raw %>%
  select(position, protein_id) %>%
  distinct() %>%
  nrow()


### Unique proteins
cp.coverage.raw %>%
  select(protein_id) %>%
  distinct() %>%
  nrow()




### Unique cysteine-fragment interactions in fragment screening df
dir_ls(paste0(path.to.data.folder, "/formatted_data"), regexp = paste0(root.date, "_figure3_fragment-all")) %>% 
  fread() %>% 
  filter(quality > 70, 
         AA == "C", 
         structure_group != "unstructured") %>%
  na.omit() %>%
  select(position, protein_id, Fragment, Dataset) %>%
  distinct() %>%
  nrow()

### Unique cys by photocaged
cp.coverage.raw %>%
  filter(str_detect(Dataset, "Caged")) %>% 
  select(protein_id, position) %>%
  distinct() %>%
  nrow()

```

